// tiktok-live-connector/dist/types/tiktok/enums.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.7.0
//   protoc               v5.27.1
// source: enums.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfitRankType = exports.MessageDisplayStyle = exports.GiftSource = exports.ExhibitionType = exports.GoalMessageSource = exports.PerceptionDialogIconType = exports.CommentTag = exports.IconDisplayType = exports.RenderType = exports.ShowType = exports.HorizontalOnclickTriggerType = exports.GiftMessageIgnoreConfig = exports.UserMetricsType = exports.TopicActionType = exports.TagType = exports.GiftMessageVersion = exports.LinkmicGiftExpressionStrategy = exports.MultiplierType = exports.PunishTypeId = exports.EmoteScene = exports.UserEmoteUploadSource = exports.RewardCondition = exports.Scene = exports.MessageType = exports.LinkMessageType = exports.CommonContentCase = exports.EnvelopeDisplay = exports.EnvelopeFollowShowStatus = exports.EnvelopeBusinessType = exports.BarrageType = exports.LinkLayerMessageType = exports.ControlAction = exports.MemberMessageAction = exports.LinkmicStatus = exports.SubscribingStatus = exports.OldSubscribeStatus = exports.SubscribeType = exports.ReplyStatus = exports.BusinessCase = exports.GroupStatus = exports.KickoutReason = exports.AgreeStatus = exports.HashtagNamespace = exports.LinkmicApplierSortSetting = exports.TextType = exports.EmotePrivateType = exports.ContentSource = exports.EmoteType = exports.AuditStatus = exports.protobufPackage = void 0;
exports.BattleInviteType = exports.BattleType = exports.SourceType = exports.TagClassification = exports.CohostLayoutMode = exports.JoinType = exports.PosIdentityType = exports.LinkmicShareRevenueSetting = exports.ContentInviteSource = exports.LinkUserType = exports.MultiGuestOutsideRoomInviteSource = exports.ContentPositionType = exports.OptPairStatus = exports.CohostABTestType = exports.PollEndType = exports.LinkmicMultiLiveEnum = exports.LinkMicUserAdminType = exports.GuestMicCameraChangeScene = exports.GuestMicCameraManageOp = exports.MuteStatus = exports.LinkRoleType = exports.LinkmicRoleType = exports.LinkSilenceStatus = exports.LinkType = exports.PlayScene = exports.BattleABTestType = exports.GiftPermissionType = exports.Result = exports.BattleAction = exports.RewardStatus = exports.BattleTaskMessageType = exports.Reason = exports.TriggerReason = exports.RankTestMessageScene = exports.PollAppealStatus = exports.PollTemplateStatus = exports.PollKind = exports.HitABStatus = exports.DisplayStyle = exports.UnionAnimationInfoType = void 0;
/* eslint-disable */
exports.protobufPackage = "";
var AuditStatus;
(function (AuditStatus) {
    AuditStatus[AuditStatus["AUDITSTATUSUNKNOWN"] = 0] = "AUDITSTATUSUNKNOWN";
    AuditStatus[AuditStatus["AUDITSTATUSPASS"] = 1] = "AUDITSTATUSPASS";
    AuditStatus[AuditStatus["AUDITSTATUSFAILED"] = 2] = "AUDITSTATUSFAILED";
    AuditStatus[AuditStatus["AUDITSTATUSREVIEWING"] = 3] = "AUDITSTATUSREVIEWING";
    AuditStatus[AuditStatus["AUDITSTATUSFORBIDDEN"] = 4] = "AUDITSTATUSFORBIDDEN";
    AuditStatus[AuditStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AuditStatus = exports.AuditStatus || (exports.AuditStatus = {}));
var EmoteType;
(function (EmoteType) {
    EmoteType[EmoteType["EMOTETYPENORMAL"] = 0] = "EMOTETYPENORMAL";
    EmoteType[EmoteType["EMOTETYPEWITHSTICKER"] = 1] = "EMOTETYPEWITHSTICKER";
    EmoteType[EmoteType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(EmoteType = exports.EmoteType || (exports.EmoteType = {}));
var ContentSource;
(function (ContentSource) {
    ContentSource[ContentSource["CONTENTSOURCEUNKNOWN"] = 0] = "CONTENTSOURCEUNKNOWN";
    ContentSource[ContentSource["CONTENTSOURCENORMAL"] = 1] = "CONTENTSOURCENORMAL";
    ContentSource[ContentSource["CONTENTSOURCECAMERA"] = 2] = "CONTENTSOURCECAMERA";
    ContentSource[ContentSource["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ContentSource = exports.ContentSource || (exports.ContentSource = {}));
var EmotePrivateType;
(function (EmotePrivateType) {
    EmotePrivateType[EmotePrivateType["EMOTE_PRIVATE_TYPE_NORMAL"] = 0] = "EMOTE_PRIVATE_TYPE_NORMAL";
    EmotePrivateType[EmotePrivateType["EMOTE_PRIVATE_TYPE_SUB_WAVE"] = 1] = "EMOTE_PRIVATE_TYPE_SUB_WAVE";
    EmotePrivateType[EmotePrivateType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(EmotePrivateType = exports.EmotePrivateType || (exports.EmotePrivateType = {}));
var TextType;
(function (TextType) {
    TextType[TextType["DISPLAY_TEXT"] = 0] = "DISPLAY_TEXT";
    TextType[TextType["CONTENT"] = 1] = "CONTENT";
    TextType[TextType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(TextType = exports.TextType || (exports.TextType = {}));
var LinkmicApplierSortSetting;
(function (LinkmicApplierSortSetting) {
    LinkmicApplierSortSetting[LinkmicApplierSortSetting["LINKMIC_APPLIER_SORT_SETTING_NONE"] = 0] = "LINKMIC_APPLIER_SORT_SETTING_NONE";
    LinkmicApplierSortSetting[LinkmicApplierSortSetting["LINKMIC_APPLIER_SORT_SETTING_BY_GIFT_SCORE"] = 1] = "LINKMIC_APPLIER_SORT_SETTING_BY_GIFT_SCORE";
    LinkmicApplierSortSetting[LinkmicApplierSortSetting["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkmicApplierSortSetting = exports.LinkmicApplierSortSetting || (exports.LinkmicApplierSortSetting = {}));
var HashtagNamespace;
(function (HashtagNamespace) {
    HashtagNamespace[HashtagNamespace["GLOBAL"] = 0] = "GLOBAL";
    HashtagNamespace[HashtagNamespace["GAMING"] = 1] = "GAMING";
    HashtagNamespace[HashtagNamespace["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(HashtagNamespace = exports.HashtagNamespace || (exports.HashtagNamespace = {}));
var AgreeStatus;
(function (AgreeStatus) {
    AgreeStatus[AgreeStatus["AGREE_UNKNOWN"] = 0] = "AGREE_UNKNOWN";
    AgreeStatus[AgreeStatus["AGREE"] = 1] = "AGREE";
    AgreeStatus[AgreeStatus["REJECT"] = 2] = "REJECT";
    AgreeStatus[AgreeStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AgreeStatus = exports.AgreeStatus || (exports.AgreeStatus = {}));
var KickoutReason;
(function (KickoutReason) {
    KickoutReason[KickoutReason["KICKOUT_REASON_UNKNOWN"] = 0] = "KICKOUT_REASON_UNKNOWN";
    KickoutReason[KickoutReason["KICKOUT_REASON_FIRST_FRAME_TIMEOUT"] = 1] = "KICKOUT_REASON_FIRST_FRAME_TIMEOUT";
    KickoutReason[KickoutReason["KICKOUT_REASON_BY_HOST"] = 2] = "KICKOUT_REASON_BY_HOST";
    KickoutReason[KickoutReason["KICKOUT_REASON_RTC_LOST_CONNECTION"] = 3] = "KICKOUT_REASON_RTC_LOST_CONNECTION";
    KickoutReason[KickoutReason["KICKOUT_REASON_BY_PUNISH"] = 4] = "KICKOUT_REASON_BY_PUNISH";
    KickoutReason[KickoutReason["KICKOUT_REASON_BY_ADMIN"] = 5] = "KICKOUT_REASON_BY_ADMIN";
    KickoutReason[KickoutReason["KICKOUT_REASON_HOST_REMOVE_ALL_GUESTS"] = 6] = "KICKOUT_REASON_HOST_REMOVE_ALL_GUESTS";
    KickoutReason[KickoutReason["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(KickoutReason = exports.KickoutReason || (exports.KickoutReason = {}));
var GroupStatus;
(function (GroupStatus) {
    GroupStatus[GroupStatus["GROUP_STATUS_UNKNOWN"] = 0] = "GROUP_STATUS_UNKNOWN";
    GroupStatus[GroupStatus["GROUP_STATUS_WAITING"] = 1] = "GROUP_STATUS_WAITING";
    GroupStatus[GroupStatus["GROUP_STATUS_LINKED"] = 3] = "GROUP_STATUS_LINKED";
    GroupStatus[GroupStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GroupStatus = exports.GroupStatus || (exports.GroupStatus = {}));
var BusinessCase;
(function (BusinessCase) {
    BusinessCase[BusinessCase["BUSINESS_NOT_SET"] = 0] = "BUSINESS_NOT_SET";
    BusinessCase[BusinessCase["APPLY_BIZ_CONTENT"] = 1] = "APPLY_BIZ_CONTENT";
    BusinessCase[BusinessCase["INVITE_BIZ_CONTENT"] = 2] = "INVITE_BIZ_CONTENT";
    BusinessCase[BusinessCase["REPLY_BIZ_CONTENT"] = 3] = "REPLY_BIZ_CONTENT";
    BusinessCase[BusinessCase["PERMIT_BIZ_CONTENT"] = 4] = "PERMIT_BIZ_CONTENT";
    BusinessCase[BusinessCase["JOIN_DIRECT_BIZ_CONTENT"] = 5] = "JOIN_DIRECT_BIZ_CONTENT";
    BusinessCase[BusinessCase["KICK_OUT_BIZ_CONTENT"] = 6] = "KICK_OUT_BIZ_CONTENT";
    BusinessCase[BusinessCase["LIST_CHANGE_BIZ_CONTENT"] = 11] = "LIST_CHANGE_BIZ_CONTENT";
    BusinessCase[BusinessCase["MULTI_LIVE_CONTENT"] = 100] = "MULTI_LIVE_CONTENT";
    BusinessCase[BusinessCase["COHOST_CONTENT"] = 200] = "COHOST_CONTENT";
    BusinessCase[BusinessCase["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BusinessCase = exports.BusinessCase || (exports.BusinessCase = {}));
var ReplyStatus;
(function (ReplyStatus) {
    ReplyStatus[ReplyStatus["REPLY_STATUS_UNKNOWN"] = 0] = "REPLY_STATUS_UNKNOWN";
    ReplyStatus[ReplyStatus["REPLY_STATUS_AGREE"] = 1] = "REPLY_STATUS_AGREE";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_PERSONALLY"] = 2] = "REPLY_STATUS_REFUSE_PERSONALLY";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_TYPE_NOT_SUPPORT"] = 3] = "REPLY_STATUS_REFUSE_TYPE_NOT_SUPPORT";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_PROCESSING_INVITATION"] = 4] = "REPLY_STATUS_REFUSE_PROCESSING_INVITATION";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_BY_TIMEOUT"] = 5] = "REPLY_STATUS_REFUSE_BY_TIMEOUT";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_EXCEPTION"] = 6] = "REPLY_STATUS_REFUSE_EXCEPTION";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_SYSTEM_NOT_SUPPORTED"] = 7] = "REPLY_STATUS_REFUSE_SYSTEM_NOT_SUPPORTED";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_SUBTYPE_DIFFERENCE"] = 8] = "REPLY_STATUS_REFUSE_SUBTYPE_DIFFERENCE";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_IN_MICROOM"] = 9] = "REPLY_STATUS_REFUSE_IN_MICROOM";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_NOT_LOAD_PLUGIN"] = 10] = "REPLY_STATUS_REFUSE_NOT_LOAD_PLUGIN";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_IN_MULTI_GUEST"] = 11] = "REPLY_STATUS_REFUSE_IN_MULTI_GUEST";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_PAUSE_LIVE"] = 12] = "REPLY_STATUS_REFUSE_PAUSE_LIVE";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_OPEN_CAMERA_DIALOG_SHOWING"] = 13] = "REPLY_STATUS_REFUSE_OPEN_CAMERA_DIALOG_SHOWING";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_DRAW_GUESSING"] = 14] = "REPLY_STATUS_REFUSE_DRAW_GUESSING";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_RANDOM_MATCHING"] = 15] = "REPLY_STATUS_REFUSE_RANDOM_MATCHING";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_IN_MATCH_PROCESSING"] = 16] = "REPLY_STATUS_REFUSE_IN_MATCH_PROCESSING";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_IN_MICROOM_FOR_MULTI_COHOST"] = 17] = "REPLY_STATUS_REFUSE_IN_MICROOM_FOR_MULTI_COHOST";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_COHOST_FINISHED"] = 18] = "REPLY_STATUS_REFUSE_COHOST_FINISHED";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_NOT_CONNECTED"] = 19] = "REPLY_STATUS_REFUSE_NOT_CONNECTED";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_LINKMIC_FULL"] = 20] = "REPLY_STATUS_REFUSE_LINKMIC_FULL";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_ARC_INCOMPATIBLE"] = 21] = "REPLY_STATUS_REFUSE_ARC_INCOMPATIBLE";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_PROCESSING_OTHER_INVITE"] = 22] = "REPLY_STATUS_REFUSE_PROCESSING_OTHER_INVITE";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_PROCESSING_OTHER_APPLY"] = 23] = "REPLY_STATUS_REFUSE_PROCESSING_OTHER_APPLY";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_IN_ANCHOR_COHOST"] = 24] = "REPLY_STATUS_REFUSE_IN_ANCHOR_COHOST";
    ReplyStatus[ReplyStatus["REPLY_STATUS_REFUSE_TOPIC_PAIRING"] = 25] = "REPLY_STATUS_REFUSE_TOPIC_PAIRING";
    ReplyStatus[ReplyStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ReplyStatus = exports.ReplyStatus || (exports.ReplyStatus = {}));
var SubscribeType;
(function (SubscribeType) {
    SubscribeType[SubscribeType["SUBSCRIBETYPE_ONCE"] = 0] = "SUBSCRIBETYPE_ONCE";
    SubscribeType[SubscribeType["SUBSCRIBETYPE_AUTO"] = 1] = "SUBSCRIBETYPE_AUTO";
    SubscribeType[SubscribeType["SUBSCRIBETYPE_DEFAULT"] = 100] = "SUBSCRIBETYPE_DEFAULT";
    SubscribeType[SubscribeType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(SubscribeType = exports.SubscribeType || (exports.SubscribeType = {}));
var OldSubscribeStatus;
(function (OldSubscribeStatus) {
    OldSubscribeStatus[OldSubscribeStatus["OLD_SUBSCRIBE_STATUS_FIRST"] = 0] = "OLD_SUBSCRIBE_STATUS_FIRST";
    OldSubscribeStatus[OldSubscribeStatus["OLD_SUBSCRIBE_STATUS_RESUB"] = 1] = "OLD_SUBSCRIBE_STATUS_RESUB";
    OldSubscribeStatus[OldSubscribeStatus["OLD_SUBSCRIBE_STATUS_SUBIN_GRACE_PERIOD"] = 2] = "OLD_SUBSCRIBE_STATUS_SUBIN_GRACE_PERIOD";
    OldSubscribeStatus[OldSubscribeStatus["OLD_SUBSCRIBE_STATUS_SUB_NOTIN_GRACE_PERIOD"] = 3] = "OLD_SUBSCRIBE_STATUS_SUB_NOTIN_GRACE_PERIOD";
    OldSubscribeStatus[OldSubscribeStatus["OLD_SUBSCRIBE_STATUS_DEFAULT"] = 100] = "OLD_SUBSCRIBE_STATUS_DEFAULT";
    OldSubscribeStatus[OldSubscribeStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(OldSubscribeStatus = exports.OldSubscribeStatus || (exports.OldSubscribeStatus = {}));
var SubscribingStatus;
(function (SubscribingStatus) {
    SubscribingStatus[SubscribingStatus["SUBSCRIBING_STATUS_UNKNOWN"] = 0] = "SUBSCRIBING_STATUS_UNKNOWN";
    SubscribingStatus[SubscribingStatus["SUBSCRIBING_STATUS_ONCE"] = 1] = "SUBSCRIBING_STATUS_ONCE";
    SubscribingStatus[SubscribingStatus["SUBSCRIBING_STATUS_CIRCLE"] = 2] = "SUBSCRIBING_STATUS_CIRCLE";
    SubscribingStatus[SubscribingStatus["SUBSCRIBING_STATUS_CIRCLE_CANCEL"] = 3] = "SUBSCRIBING_STATUS_CIRCLE_CANCEL";
    SubscribingStatus[SubscribingStatus["SUBSCRIBING_STATUS_REFUND"] = 4] = "SUBSCRIBING_STATUS_REFUND";
    SubscribingStatus[SubscribingStatus["SUBSCRIBING_STATUS_IN_GRACE_PERIOD"] = 5] = "SUBSCRIBING_STATUS_IN_GRACE_PERIOD";
    SubscribingStatus[SubscribingStatus["SUBSCRIBING_STATUS_NOT_IN_GRACE_PERIOD"] = 6] = "SUBSCRIBING_STATUS_NOT_IN_GRACE_PERIOD";
    SubscribingStatus[SubscribingStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(SubscribingStatus = exports.SubscribingStatus || (exports.SubscribingStatus = {}));
var LinkmicStatus;
(function (LinkmicStatus) {
    LinkmicStatus[LinkmicStatus["Disable"] = 0] = "Disable";
    LinkmicStatus[LinkmicStatus["Enable"] = 1] = "Enable";
    LinkmicStatus[LinkmicStatus["Just_Following"] = 2] = "Just_Following";
    LinkmicStatus[LinkmicStatus["Multi_Linking"] = 3] = "Multi_Linking";
    LinkmicStatus[LinkmicStatus["Multi_Linking_Only_Following"] = 4] = "Multi_Linking_Only_Following";
    LinkmicStatus[LinkmicStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkmicStatus = exports.LinkmicStatus || (exports.LinkmicStatus = {}));
var MemberMessageAction;
(function (MemberMessageAction) {
    MemberMessageAction[MemberMessageAction["UNKNOWN"] = 0] = "UNKNOWN";
    /** JOINED - User Joined the Stream */
    MemberMessageAction[MemberMessageAction["JOINED"] = 1] = "JOINED";
    /** SUBSCRIBED - User Subscribed to the Host */
    MemberMessageAction[MemberMessageAction["SUBSCRIBED"] = 3] = "SUBSCRIBED";
    MemberMessageAction[MemberMessageAction["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(MemberMessageAction = exports.MemberMessageAction || (exports.MemberMessageAction = {}));
var ControlAction;
(function (ControlAction) {
    ControlAction[ControlAction["CONTROL_ACTION_FALLBACK_UNKNOWN"] = 0] = "CONTROL_ACTION_FALLBACK_UNKNOWN";
    ControlAction[ControlAction["CONTROL_ACTION_STREAM_PAUSED"] = 1] = "CONTROL_ACTION_STREAM_PAUSED";
    ControlAction[ControlAction["CONTROL_ACTION_STREAM_UNPAUSED"] = 2] = "CONTROL_ACTION_STREAM_UNPAUSED";
    ControlAction[ControlAction["CONTROL_ACTION_STREAM_ENDED"] = 3] = "CONTROL_ACTION_STREAM_ENDED";
    ControlAction[ControlAction["CONTROL_ACTION_STREAM_SUSPENDED"] = 4] = "CONTROL_ACTION_STREAM_SUSPENDED";
    ControlAction[ControlAction["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ControlAction = exports.ControlAction || (exports.ControlAction = {}));
var LinkLayerMessageType;
(function (LinkLayerMessageType) {
    LinkLayerMessageType[LinkLayerMessageType["Linker_Unknown"] = 0] = "Linker_Unknown";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Create"] = 1] = "Linker_Create";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Invite"] = 2] = "Linker_Invite";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Apply"] = 3] = "Linker_Apply";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Permit"] = 4] = "Linker_Permit";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Reply"] = 5] = "Linker_Reply";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Kick_Out"] = 6] = "Linker_Kick_Out";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Cancel_Apply"] = 7] = "Linker_Cancel_Apply";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Cancel_Invite"] = 8] = "Linker_Cancel_Invite";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Leave"] = 9] = "Linker_Leave";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Finish"] = 10] = "Linker_Finish";
    LinkLayerMessageType[LinkLayerMessageType["Linker_List_Change"] = 11] = "Linker_List_Change";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Join_Direct"] = 12] = "Linker_Join_Direct";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Join_Group"] = 13] = "Linker_Join_Group";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Permit_Group"] = 14] = "Linker_Permit_Group";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Cancel_Group"] = 15] = "Linker_Cancel_Group";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Leave_Group"] = 16] = "Linker_Leave_Group";
    LinkLayerMessageType[LinkLayerMessageType["Linker_P2P_Group_Change"] = 17] = "Linker_P2P_Group_Change";
    LinkLayerMessageType[LinkLayerMessageType["Linker_Group_Change"] = 18] = "Linker_Group_Change";
    LinkLayerMessageType[LinkLayerMessageType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkLayerMessageType = exports.LinkLayerMessageType || (exports.LinkLayerMessageType = {}));
var BarrageType;
(function (BarrageType) {
    BarrageType[BarrageType["BarrageType_Unknown"] = 0] = "BarrageType_Unknown";
    BarrageType[BarrageType["EComOrdering"] = 1] = "EComOrdering";
    BarrageType[BarrageType["EComBuying"] = 2] = "EComBuying";
    BarrageType[BarrageType["Normal"] = 3] = "Normal";
    BarrageType[BarrageType["Subscribe"] = 4] = "Subscribe";
    BarrageType[BarrageType["EventView"] = 5] = "EventView";
    BarrageType[BarrageType["EventRegistered"] = 6] = "EventRegistered";
    BarrageType[BarrageType["SubscribeGift"] = 7] = "SubscribeGift";
    BarrageType[BarrageType["UserUpgrade"] = 8] = "UserUpgrade";
    BarrageType[BarrageType["GradeUserEntranceNotification"] = 9] = "GradeUserEntranceNotification";
    BarrageType[BarrageType["FansLevelUpgrade"] = 10] = "FansLevelUpgrade";
    BarrageType[BarrageType["FansLevelEntrance"] = 11] = "FansLevelEntrance";
    BarrageType[BarrageType["GamePartnership"] = 12] = "GamePartnership";
    BarrageType[BarrageType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BarrageType = exports.BarrageType || (exports.BarrageType = {}));
var EnvelopeBusinessType;
(function (EnvelopeBusinessType) {
    EnvelopeBusinessType[EnvelopeBusinessType["BusinessTypeUnknown"] = 0] = "BusinessTypeUnknown";
    EnvelopeBusinessType[EnvelopeBusinessType["BusinessTypeUserDiamond"] = 1] = "BusinessTypeUserDiamond";
    EnvelopeBusinessType[EnvelopeBusinessType["BusinessTypePlatformDiamond"] = 2] = "BusinessTypePlatformDiamond";
    EnvelopeBusinessType[EnvelopeBusinessType["BusinessTypePlatformShell"] = 3] = "BusinessTypePlatformShell";
    EnvelopeBusinessType[EnvelopeBusinessType["BusinessTypePortal"] = 4] = "BusinessTypePortal";
    EnvelopeBusinessType[EnvelopeBusinessType["BusinessTypePlatformMerch"] = 5] = "BusinessTypePlatformMerch";
    EnvelopeBusinessType[EnvelopeBusinessType["BusinessTypeEoYDiamond"] = 6] = "BusinessTypeEoYDiamond";
    EnvelopeBusinessType[EnvelopeBusinessType["BusinessTypeFanClubGtM"] = 7] = "BusinessTypeFanClubGtM";
    EnvelopeBusinessType[EnvelopeBusinessType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(EnvelopeBusinessType = exports.EnvelopeBusinessType || (exports.EnvelopeBusinessType = {}));
var EnvelopeFollowShowStatus;
(function (EnvelopeFollowShowStatus) {
    EnvelopeFollowShowStatus[EnvelopeFollowShowStatus["ENVELOPE_FOLLOW_SHOW_STATUS_ENVELOPE_FOLLOW_SHOW_UNKNOWN"] = 0] = "ENVELOPE_FOLLOW_SHOW_STATUS_ENVELOPE_FOLLOW_SHOW_UNKNOWN";
    EnvelopeFollowShowStatus[EnvelopeFollowShowStatus["ENVELOPE_FOLLOW_SHOW_STATUS_ENVELOPE_FOLLOW_SHOW"] = 1] = "ENVELOPE_FOLLOW_SHOW_STATUS_ENVELOPE_FOLLOW_SHOW";
    EnvelopeFollowShowStatus[EnvelopeFollowShowStatus["ENVELOPE_FOLLOW_SHOW_STATUS_ENVELOPE_FOLLOW_NOT_SHOW"] = 2] = "ENVELOPE_FOLLOW_SHOW_STATUS_ENVELOPE_FOLLOW_NOT_SHOW";
    EnvelopeFollowShowStatus[EnvelopeFollowShowStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(EnvelopeFollowShowStatus = exports.EnvelopeFollowShowStatus || (exports.EnvelopeFollowShowStatus = {}));
var EnvelopeDisplay;
(function (EnvelopeDisplay) {
    EnvelopeDisplay[EnvelopeDisplay["ENVELOPE_DISPLAY_UNKNOWN"] = 0] = "ENVELOPE_DISPLAY_UNKNOWN";
    EnvelopeDisplay[EnvelopeDisplay["ENVELOPE_DISPLAY_NEW"] = 1] = "ENVELOPE_DISPLAY_NEW";
    EnvelopeDisplay[EnvelopeDisplay["ENVELOPE_DISPLAY_HIDE"] = 2] = "ENVELOPE_DISPLAY_HIDE";
    EnvelopeDisplay[EnvelopeDisplay["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(EnvelopeDisplay = exports.EnvelopeDisplay || (exports.EnvelopeDisplay = {}));
var CommonContentCase;
(function (CommonContentCase) {
    CommonContentCase[CommonContentCase["COMMON_CONTENT_NOT_SET"] = 0] = "COMMON_CONTENT_NOT_SET";
    CommonContentCase[CommonContentCase["CREATE_CHANNEL_CONTENT"] = 100] = "CREATE_CHANNEL_CONTENT";
    CommonContentCase[CommonContentCase["LIST_CHANGE_CONTENT"] = 102] = "LIST_CHANGE_CONTENT";
    CommonContentCase[CommonContentCase["INVITE_CONTENT"] = 103] = "INVITE_CONTENT";
    CommonContentCase[CommonContentCase["APPLY_CONTENT"] = 104] = "APPLY_CONTENT";
    CommonContentCase[CommonContentCase["PERMIT_APPLY_CONTENT"] = 105] = "PERMIT_APPLY_CONTENT";
    CommonContentCase[CommonContentCase["REPLY_INVITE_CONTENT"] = 106] = "REPLY_INVITE_CONTENT";
    CommonContentCase[CommonContentCase["KICK_OUT_CONTENT"] = 107] = "KICK_OUT_CONTENT";
    CommonContentCase[CommonContentCase["CANCEL_APPLY_CONTENT"] = 108] = "CANCEL_APPLY_CONTENT";
    CommonContentCase[CommonContentCase["CANCEL_INVITE_CONTENT"] = 109] = "CANCEL_INVITE_CONTENT";
    CommonContentCase[CommonContentCase["LEAVE_CONTENT"] = 110] = "LEAVE_CONTENT";
    CommonContentCase[CommonContentCase["FINISH_CONTENT"] = 111] = "FINISH_CONTENT";
    CommonContentCase[CommonContentCase["JOIN_DIRECT_CONTENT"] = 112] = "JOIN_DIRECT_CONTENT";
    CommonContentCase[CommonContentCase["JOIN_GROUP_CONTENT"] = 113] = "JOIN_GROUP_CONTENT";
    CommonContentCase[CommonContentCase["PERMIT_GROUP_CONTENT"] = 114] = "PERMIT_GROUP_CONTENT";
    CommonContentCase[CommonContentCase["CANCEL_GROUP_CONTENT"] = 115] = "CANCEL_GROUP_CONTENT";
    CommonContentCase[CommonContentCase["LEAVE_GROUP_CONTENT"] = 116] = "LEAVE_GROUP_CONTENT";
    CommonContentCase[CommonContentCase["P2P_GROUP_CHANGE_CONTENT"] = 117] = "P2P_GROUP_CHANGE_CONTENT";
    CommonContentCase[CommonContentCase["GROUP_CHANGE_CONTENT"] = 118] = "GROUP_CHANGE_CONTENT";
    CommonContentCase[CommonContentCase["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(CommonContentCase = exports.CommonContentCase || (exports.CommonContentCase = {}));
var LinkMessageType;
(function (LinkMessageType) {
    LinkMessageType[LinkMessageType["TYPE_LINKER_UNKNOWN"] = 0] = "TYPE_LINKER_UNKNOWN";
    LinkMessageType[LinkMessageType["TYPE_LINKER_CREATE"] = 1] = "TYPE_LINKER_CREATE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_CLOSE"] = 2] = "TYPE_LINKER_CLOSE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_INVITE"] = 3] = "TYPE_LINKER_INVITE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_APPLY"] = 4] = "TYPE_LINKER_APPLY";
    LinkMessageType[LinkMessageType["TYPE_LINKER_REPLY"] = 5] = "TYPE_LINKER_REPLY";
    LinkMessageType[LinkMessageType["TYPE_LINKER_ENTER"] = 6] = "TYPE_LINKER_ENTER";
    LinkMessageType[LinkMessageType["TYPE_LINKER_LEAVE"] = 7] = "TYPE_LINKER_LEAVE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_PERMIT"] = 8] = "TYPE_LINKER_PERMIT";
    LinkMessageType[LinkMessageType["TYPE_LINKER_CANCEL_INVITE"] = 9] = "TYPE_LINKER_CANCEL_INVITE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_WAITING_LIST_CHANGE"] = 10] = "TYPE_LINKER_WAITING_LIST_CHANGE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_LINKED_LIST_CHANGE"] = 11] = "TYPE_LINKER_LINKED_LIST_CHANGE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_UPDATE_USER"] = 12] = "TYPE_LINKER_UPDATE_USER";
    LinkMessageType[LinkMessageType["TYPE_LINKER_KICK_OUT"] = 13] = "TYPE_LINKER_KICK_OUT";
    LinkMessageType[LinkMessageType["TYPE_LINKER_CANCEL_APPLY"] = 14] = "TYPE_LINKER_CANCEL_APPLY";
    LinkMessageType[LinkMessageType["TYPE_LINKER_MUTE"] = 15] = "TYPE_LINKER_MUTE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_MATCH"] = 16] = "TYPE_LINKER_MATCH";
    LinkMessageType[LinkMessageType["TYPE_LINKER_UPDATE_USER_SETTING"] = 17] = "TYPE_LINKER_UPDATE_USER_SETTING";
    LinkMessageType[LinkMessageType["TYPE_LINKER_MIC_IDX_UPDATE"] = 18] = "TYPE_LINKER_MIC_IDX_UPDATE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_LEAVE_V2"] = 19] = "TYPE_LINKER_LEAVE_V2";
    LinkMessageType[LinkMessageType["TYPE_LINKER_WAITING_LIST_CHANGE_V2"] = 20] = "TYPE_LINKER_WAITING_LIST_CHANGE_V2";
    LinkMessageType[LinkMessageType["TYPE_LINKER_LINKED_LIST_CHANGE_V2"] = 21] = "TYPE_LINKER_LINKED_LIST_CHANGE_V2";
    LinkMessageType[LinkMessageType["TYPE_LINKER_COHOST_LIST_CHANGE"] = 22] = "TYPE_LINKER_COHOST_LIST_CHANGE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_MEDIA_CHANGE"] = 23] = "TYPE_LINKER_MEDIA_CHANGE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_ACCEPT_NOTICE"] = 24] = "TYPE_LINKER_ACCEPT_NOTICE";
    LinkMessageType[LinkMessageType["TYPE_LINKER_SYS_KICK_OUT"] = 101] = "TYPE_LINKER_SYS_KICK_OUT";
    LinkMessageType[LinkMessageType["TYPE_LINKMIC_USER_TOAST"] = 102] = "TYPE_LINKMIC_USER_TOAST";
    LinkMessageType[LinkMessageType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkMessageType = exports.LinkMessageType || (exports.LinkMessageType = {}));
var MessageType;
(function (MessageType) {
    MessageType[MessageType["MESSAGE_TYPE_SUB_SUCCESS"] = 0] = "MESSAGE_TYPE_SUB_SUCCESS";
    MessageType[MessageType["MESSAGE_TYPE_ANCHOR_REMINDER"] = 1] = "MESSAGE_TYPE_ANCHOR_REMINDER";
    MessageType[MessageType["MESSAGE_TYPE_ENTER_ROOM_EXPIRE_SOON"] = 2] = "MESSAGE_TYPE_ENTER_ROOM_EXPIRE_SOON";
    MessageType[MessageType["MESSAGE_TYPE_SUB_GOAL_CREATE_TO_ANCHOR"] = 3] = "MESSAGE_TYPE_SUB_GOAL_CREATE_TO_ANCHOR";
    MessageType[MessageType["MESSAGE_TYPE_SUB_GOAL_COMPLETE_TO_AUDIENCE"] = 4] = "MESSAGE_TYPE_SUB_GOAL_COMPLETE_TO_AUDIENCE";
    MessageType[MessageType["MESSAGE_TYPE_SUB_GOAL_COMPLETE_TO_ANCHOR"] = 5] = "MESSAGE_TYPE_SUB_GOAL_COMPLETE_TO_ANCHOR";
    MessageType[MessageType["MESSAGE_TYPE_SUB_GIFT_TIK_TOK_2_USER_NOTICE"] = 6] = "MESSAGE_TYPE_SUB_GIFT_TIK_TOK_2_USER_NOTICE";
    MessageType[MessageType["MESSAGE_TYPE_SUB_GIFT_TIK_TOK_2_ANCHOR_NOTICE"] = 7] = "MESSAGE_TYPE_SUB_GIFT_TIK_TOK_2_ANCHOR_NOTICE";
    MessageType[MessageType["MESSAGE_TYPE_SUB_GIFT_T_RECEIVES_END_NOTICE"] = 8] = "MESSAGE_TYPE_SUB_GIFT_T_RECEIVES_END_NOTICE";
    MessageType[MessageType["MESSAGE_TYPE_SUB_GIFTS_END_SUCCEED_ROOM_MESSAGE"] = 9] = "MESSAGE_TYPE_SUB_GIFTS_END_SUCCEED_ROOM_MESSAGE";
    MessageType[MessageType["MESSAGE_TYPE_SUB_GIFTS_END_SUCCEED_ANCHOR_NOTICE"] = 10] = "MESSAGE_TYPE_SUB_GIFTS_END_SUCCEED_ANCHOR_NOTICE";
    MessageType[MessageType["MESSAGE_TYPE_SUB_GIFT_LOW_VERSION_UPGRADE_NOTICE"] = 11] = "MESSAGE_TYPE_SUB_GIFT_LOW_VERSION_UPGRADE_NOTICE";
    MessageType[MessageType["MESSAGE_TYPE_SUB_GIFT_USER_BUY_AU_TH_NOTICE"] = 12] = "MESSAGE_TYPE_SUB_GIFT_USER_BUY_AU_TH_NOTICE";
    MessageType[MessageType["MESSAGE_TYPE_SUB_COMMON_TEXT_NOTICE"] = 13] = "MESSAGE_TYPE_SUB_COMMON_TEXT_NOTICE";
    MessageType[MessageType["MESSAGE_TYPE_SUB_MODERATOR_PIN_PERK"] = 14] = "MESSAGE_TYPE_SUB_MODERATOR_PIN_PERK";
    MessageType[MessageType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(MessageType = exports.MessageType || (exports.MessageType = {}));
var Scene;
(function (Scene) {
    Scene[Scene["SCENE_UNKNOWN"] = 0] = "SCENE_UNKNOWN";
    Scene[Scene["SCENE_CO_HOST"] = 2] = "SCENE_CO_HOST";
    Scene[Scene["SCENE_MULTI_LIVE"] = 4] = "SCENE_MULTI_LIVE";
    Scene[Scene["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Scene = exports.Scene || (exports.Scene = {}));
var RewardCondition;
(function (RewardCondition) {
    RewardCondition[RewardCondition["REWARD_CONDITION_SUBSCRIPTION"] = 0] = "REWARD_CONDITION_SUBSCRIPTION";
    RewardCondition[RewardCondition["REWARD_CONDITION_SUB_WAVE_CUSTOM"] = 1] = "REWARD_CONDITION_SUB_WAVE_CUSTOM";
    RewardCondition[RewardCondition["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(RewardCondition = exports.RewardCondition || (exports.RewardCondition = {}));
var UserEmoteUploadSource;
(function (UserEmoteUploadSource) {
    UserEmoteUploadSource[UserEmoteUploadSource["USER_EMOTE_UPLOAD_SOURCE_EMOTE_UPLOAD_SOURCE_ANCHOR"] = 0] = "USER_EMOTE_UPLOAD_SOURCE_EMOTE_UPLOAD_SOURCE_ANCHOR";
    UserEmoteUploadSource[UserEmoteUploadSource["USER_EMOTE_UPLOAD_SOURCE_EMOTE_UPLOAD_SOURCE_SUBSCRIBER"] = 1] = "USER_EMOTE_UPLOAD_SOURCE_EMOTE_UPLOAD_SOURCE_SUBSCRIBER";
    UserEmoteUploadSource[UserEmoteUploadSource["USER_EMOTE_UPLOAD_SOURCE_EMOTE_UPLOAD_SOURCE_MODERATOR"] = 2] = "USER_EMOTE_UPLOAD_SOURCE_EMOTE_UPLOAD_SOURCE_MODERATOR";
    UserEmoteUploadSource[UserEmoteUploadSource["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(UserEmoteUploadSource = exports.UserEmoteUploadSource || (exports.UserEmoteUploadSource = {}));
var EmoteScene;
(function (EmoteScene) {
    EmoteScene[EmoteScene["EMOTE_SCENE_SUBSCRIPTION"] = 0] = "EMOTE_SCENE_SUBSCRIPTION";
    EmoteScene[EmoteScene["EMOTE_SCENE_GAME"] = 1] = "EMOTE_SCENE_GAME";
    EmoteScene[EmoteScene["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(EmoteScene = exports.EmoteScene || (exports.EmoteScene = {}));
var PunishTypeId;
(function (PunishTypeId) {
    PunishTypeId[PunishTypeId["PUNISH_TYPE_IDUN_KNOWN"] = 0] = "PUNISH_TYPE_IDUN_KNOWN";
    PunishTypeId[PunishTypeId["PUNISH_TYPE_ID_BAN_LINK_MIC"] = 9] = "PUNISH_TYPE_ID_BAN_LINK_MIC";
    PunishTypeId[PunishTypeId["PUNISH_TYPE_ID_BAN_GAME_PARTNERSHIP"] = 25] = "PUNISH_TYPE_ID_BAN_GAME_PARTNERSHIP";
    PunishTypeId[PunishTypeId["PUNISH_TYPE_ID_REMOVE_GAME_PARTNERSHIP"] = 26] = "PUNISH_TYPE_ID_REMOVE_GAME_PARTNERSHIP";
    PunishTypeId[PunishTypeId["PUNISH_TYPE_ID_BANCO_HOST_LINK_MIC"] = 55] = "PUNISH_TYPE_ID_BANCO_HOST_LINK_MIC";
    PunishTypeId[PunishTypeId["PUNISH_TYPE_ID_AUTHORITY_LIMIT_MATCH"] = 57] = "PUNISH_TYPE_ID_AUTHORITY_LIMIT_MATCH";
    PunishTypeId[PunishTypeId["PUNISH_TYPE_ID_BAN_VOICE_CHAT"] = 59] = "PUNISH_TYPE_ID_BAN_VOICE_CHAT";
    PunishTypeId[PunishTypeId["PUNISH_TYPE_ID_BAN_LIVE_GOAL"] = 64] = "PUNISH_TYPE_ID_BAN_LIVE_GOAL";
    PunishTypeId[PunishTypeId["PUNISH_TYPE_ID_VIEWER_LIMIT"] = 70] = "PUNISH_TYPE_ID_VIEWER_LIMIT";
    PunishTypeId[PunishTypeId["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PunishTypeId = exports.PunishTypeId || (exports.PunishTypeId = {}));
var MultiplierType;
(function (MultiplierType) {
    MultiplierType[MultiplierType["MULTIPLIER_TYPE_UNSPECIFIED"] = 0] = "MULTIPLIER_TYPE_UNSPECIFIED";
    MultiplierType[MultiplierType["MULTIPLIER_TYPE_CRITICAL_STRIKE"] = 1] = "MULTIPLIER_TYPE_CRITICAL_STRIKE";
    MultiplierType[MultiplierType["MULTIPLIER_TYPE_TOP_2"] = 2] = "MULTIPLIER_TYPE_TOP_2";
    MultiplierType[MultiplierType["MULTIPLIER_TYPE_TOP_3"] = 3] = "MULTIPLIER_TYPE_TOP_3";
    MultiplierType[MultiplierType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(MultiplierType = exports.MultiplierType || (exports.MultiplierType = {}));
var LinkmicGiftExpressionStrategy;
(function (LinkmicGiftExpressionStrategy) {
    LinkmicGiftExpressionStrategy[LinkmicGiftExpressionStrategy["LINKMIC_GIFT_EXPRESSION_STRATEGY_CONTROL_V_1"] = 0] = "LINKMIC_GIFT_EXPRESSION_STRATEGY_CONTROL_V_1";
    LinkmicGiftExpressionStrategy[LinkmicGiftExpressionStrategy["LINKMIC_GIFT_EXPRESSION_STRATEGY_EXPERIMENT_V_1"] = 1] = "LINKMIC_GIFT_EXPRESSION_STRATEGY_EXPERIMENT_V_1";
    LinkmicGiftExpressionStrategy[LinkmicGiftExpressionStrategy["LINKMIC_GIFT_EXPRESSION_STRATEGY_EXPERIMENT_V_2"] = 2] = "LINKMIC_GIFT_EXPRESSION_STRATEGY_EXPERIMENT_V_2";
    LinkmicGiftExpressionStrategy[LinkmicGiftExpressionStrategy["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkmicGiftExpressionStrategy = exports.LinkmicGiftExpressionStrategy || (exports.LinkmicGiftExpressionStrategy = {}));
var GiftMessageVersion;
(function (GiftMessageVersion) {
    GiftMessageVersion[GiftMessageVersion["GIFT_MESSAGE_VERSION_0"] = 0] = "GIFT_MESSAGE_VERSION_0";
    GiftMessageVersion[GiftMessageVersion["GIFT_MESSAGE_VERSION_1"] = 1] = "GIFT_MESSAGE_VERSION_1";
    GiftMessageVersion[GiftMessageVersion["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GiftMessageVersion = exports.GiftMessageVersion || (exports.GiftMessageVersion = {}));
var TagType;
(function (TagType) {
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_UNKNOWN"] = 0] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_UNKNOWN";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_USER_GRADE"] = 1] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_USER_GRADE";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FANS_LEVEL"] = 2] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FANS_LEVEL";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_WATCH_ME_DAYS_AGO"] = 3] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_WATCH_ME_DAYS_AGO";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_CUSTOM"] = 4] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_CUSTOM";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_TITLE_GIFT"] = 5] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_TITLE_GIFT";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FIRST_JOINED_TEAM"] = 6] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FIRST_JOINED_TEAM";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_PAY_ACCOMPANY_DAYS"] = 7] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_PAY_ACCOMPANY_DAYS";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_SPONSOR_GIFT_LAST_ROOM"] = 8] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_SPONSOR_GIFT_LAST_ROOM";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_MATCH_MVP_LAST_ROOM"] = 9] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_MATCH_MVP_LAST_ROOM";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_LARGE_AMOUNT_GIFT_LAST_ROOM"] = 10] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_LARGE_AMOUNT_GIFT_LAST_ROOM";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_COMMENT_LAST_ROOM"] = 11] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_COMMENT_LAST_ROOM";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_TITLED_GIFT"] = 12] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_TITLED_GIFT";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_MEET_ANNIVERSARY"] = 13] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_MEET_ANNIVERSARY";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FANS_SLEEP"] = 14] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FANS_SLEEP";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_NOT_SEND_HEART_ME"] = 15] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_NOT_SEND_HEART_ME";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_NOT_JOIN_TEAM"] = 16] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_NOT_JOIN_TEAM";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FIRST_WATCH_LIVE"] = 17] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FIRST_WATCH_LIVE";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_COMMENT"] = 18] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_COMMENT";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_GIFT_TIMES"] = 19] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_GIFT_TIMES";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_WATCH_LIVE_DURATION"] = 20] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_WATCH_LIVE_DURATION";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_GIFT"] = 21] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_GIFT";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_LIVE_CONTRIBUTION_TOP"] = 22] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_RECENT_LIVE_CONTRIBUTION_TOP";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_JUST_UPGRADE"] = 28] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_JUST_UPGRADE";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FAN_TOTAL_WATCH_DURATION"] = 29] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FAN_TOTAL_WATCH_DURATION";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FAN_TOTAL_COMMENT_NUM"] = 30] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FAN_TOTAL_COMMENT_NUM";
    TagType[TagType["TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FAN_TOTAL_GIFT_SENT_NUM"] = 31] = "TAG_TYPE_CREATOR_CR_M_TAG_TYPE_FAN_TOTAL_GIFT_SENT_NUM";
    TagType[TagType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(TagType = exports.TagType || (exports.TagType = {}));
var TopicActionType;
(function (TopicActionType) {
    TopicActionType[TopicActionType["TOPIC_ACTION_TYPE_UNKNOWN"] = 0] = "TOPIC_ACTION_TYPE_UNKNOWN";
    TopicActionType[TopicActionType["TOPIC_ACTION_TYPE_FOLLOW"] = 1] = "TOPIC_ACTION_TYPE_FOLLOW";
    TopicActionType[TopicActionType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(TopicActionType = exports.TopicActionType || (exports.TopicActionType = {}));
var UserMetricsType;
(function (UserMetricsType) {
    UserMetricsType[UserMetricsType["USER_METRICS_TYPE_UNKNOWN"] = 0] = "USER_METRICS_TYPE_UNKNOWN";
    UserMetricsType[UserMetricsType["USER_METRICS_TYPE_GRADE"] = 1] = "USER_METRICS_TYPE_GRADE";
    UserMetricsType[UserMetricsType["USER_METRICS_TYPE_SUBSCRIBE"] = 2] = "USER_METRICS_TYPE_SUBSCRIBE";
    UserMetricsType[UserMetricsType["USER_METRICS_TYPE_FOLLOW"] = 3] = "USER_METRICS_TYPE_FOLLOW";
    UserMetricsType[UserMetricsType["USER_METRICS_TYPE_FANS_CLUB"] = 4] = "USER_METRICS_TYPE_FANS_CLUB";
    UserMetricsType[UserMetricsType["USER_METRICS_TYPE_TOP_VIEWER"] = 5] = "USER_METRICS_TYPE_TOP_VIEWER";
    UserMetricsType[UserMetricsType["USER_METRICS_TYPE_GIFT"] = 6] = "USER_METRICS_TYPE_GIFT";
    UserMetricsType[UserMetricsType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(UserMetricsType = exports.UserMetricsType || (exports.UserMetricsType = {}));
var GiftMessageIgnoreConfig;
(function (GiftMessageIgnoreConfig) {
    GiftMessageIgnoreConfig[GiftMessageIgnoreConfig["GIFT_MESSAGE_IGNORE_CONFIG_NOT_IGNORE"] = 0] = "GIFT_MESSAGE_IGNORE_CONFIG_NOT_IGNORE";
    GiftMessageIgnoreConfig[GiftMessageIgnoreConfig["GIFT_MESSAGE_IGNORE_CONFIG_IGNORE_TRAY"] = 1] = "GIFT_MESSAGE_IGNORE_CONFIG_IGNORE_TRAY";
    GiftMessageIgnoreConfig[GiftMessageIgnoreConfig["GIFT_MESSAGE_IGNORE_CONFIG_IGNORE_PS_M"] = 2] = "GIFT_MESSAGE_IGNORE_CONFIG_IGNORE_PS_M";
    GiftMessageIgnoreConfig[GiftMessageIgnoreConfig["GIFT_MESSAGE_IGNORE_CONFIG_IGNORE_TRAY_AND_PS_M"] = 3] = "GIFT_MESSAGE_IGNORE_CONFIG_IGNORE_TRAY_AND_PS_M";
    GiftMessageIgnoreConfig[GiftMessageIgnoreConfig["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GiftMessageIgnoreConfig = exports.GiftMessageIgnoreConfig || (exports.GiftMessageIgnoreConfig = {}));
var HorizontalOnclickTriggerType;
(function (HorizontalOnclickTriggerType) {
    HorizontalOnclickTriggerType[HorizontalOnclickTriggerType["HORIZONTAL_ONCLICK_TRIGGER_TYPE_ONCLICK_TRIGGER_TYPE_UNKNOWN"] = 0] = "HORIZONTAL_ONCLICK_TRIGGER_TYPE_ONCLICK_TRIGGER_TYPE_UNKNOWN";
    HorizontalOnclickTriggerType[HorizontalOnclickTriggerType["HORIZONTAL_ONCLICK_TRIGGER_TYPE_ONCLICK_TRIGGER_TYPE_LEFT"] = 1] = "HORIZONTAL_ONCLICK_TRIGGER_TYPE_ONCLICK_TRIGGER_TYPE_LEFT";
    HorizontalOnclickTriggerType[HorizontalOnclickTriggerType["HORIZONTAL_ONCLICK_TRIGGER_TYPE_ONCLICK_TRIGGER_TYPE_MIDDLE"] = 2] = "HORIZONTAL_ONCLICK_TRIGGER_TYPE_ONCLICK_TRIGGER_TYPE_MIDDLE";
    HorizontalOnclickTriggerType[HorizontalOnclickTriggerType["HORIZONTAL_ONCLICK_TRIGGER_TYPE_ONCLICK_TRIGGER_TYPE_RIGHT"] = 3] = "HORIZONTAL_ONCLICK_TRIGGER_TYPE_ONCLICK_TRIGGER_TYPE_RIGHT";
    HorizontalOnclickTriggerType[HorizontalOnclickTriggerType["HORIZONTAL_ONCLICK_TRIGGER_TYPE_ONCLICK_TRIGGER_TYPE_ALL_AREA"] = 4] = "HORIZONTAL_ONCLICK_TRIGGER_TYPE_ONCLICK_TRIGGER_TYPE_ALL_AREA";
    HorizontalOnclickTriggerType[HorizontalOnclickTriggerType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(HorizontalOnclickTriggerType = exports.HorizontalOnclickTriggerType || (exports.HorizontalOnclickTriggerType = {}));
var ShowType;
(function (ShowType) {
    ShowType[ShowType["SHOW_TYPE_NORMAL"] = 0] = "SHOW_TYPE_NORMAL";
    ShowType[ShowType["SHOW_TYPE_FADE_IN_OUT"] = 1] = "SHOW_TYPE_FADE_IN_OUT";
    ShowType[ShowType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ShowType = exports.ShowType || (exports.ShowType = {}));
var RenderType;
(function (RenderType) {
    RenderType[RenderType["RENDER_TYPE_NATIVE"] = 0] = "RENDER_TYPE_NATIVE";
    RenderType[RenderType["RENDER_TYPE_HYBRID"] = 1] = "RENDER_TYPE_HYBRID";
    RenderType[RenderType["RENDER_TYPE_ALPHA"] = 2] = "RENDER_TYPE_ALPHA";
    RenderType[RenderType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(RenderType = exports.RenderType || (exports.RenderType = {}));
var IconDisplayType;
(function (IconDisplayType) {
    IconDisplayType[IconDisplayType["ICON_DISPLAY_TYPE_IMAGE"] = 0] = "ICON_DISPLAY_TYPE_IMAGE";
    IconDisplayType[IconDisplayType["ICON_DISPLAY_TYPE_BADGE"] = 1] = "ICON_DISPLAY_TYPE_BADGE";
    IconDisplayType[IconDisplayType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(IconDisplayType = exports.IconDisplayType || (exports.IconDisplayType = {}));
var CommentTag;
(function (CommentTag) {
    CommentTag[CommentTag["COMMENT_TAG_NORMAL"] = 0] = "COMMENT_TAG_NORMAL";
    CommentTag[CommentTag["COMMENT_TAG_CANDIDATE"] = 1] = "COMMENT_TAG_CANDIDATE";
    CommentTag[CommentTag["COMMENT_TAG_OVERAGE"] = 2] = "COMMENT_TAG_OVERAGE";
    CommentTag[CommentTag["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(CommentTag = exports.CommentTag || (exports.CommentTag = {}));
var PerceptionDialogIconType;
(function (PerceptionDialogIconType) {
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_NONE"] = 0] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_NONE";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_WARNING"] = 1] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_WARNING";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_LINK_MIC"] = 2] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_LINK_MIC";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_GUEST_LINK_MIC"] = 3] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_GUEST_LINK_MIC";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_LIVE"] = 4] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_LIVE";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_TREASURE_BOX"] = 5] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_TREASURE_BOX";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_MUTE"] = 6] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_MUTE";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_GAMEPAD_ACCESS_REVOKED"] = 7] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_GAMEPAD_ACCESS_REVOKED";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_BAN_REPORT_LIVE_SINGLE_ROOM"] = 8] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_BAN_REPORT_LIVE_SINGLE_ROOM";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_BAN_REPORT_LIVE_ALL_ROOM"] = 9] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_BAN_REPORT_LIVE_ALL_ROOM";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_BAN_REPORT_LIVE_GREEN_SCREEN"] = 10] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_BAN_REPORT_LIVE_GREEN_SCREEN";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_GIFT"] = 11] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_GIFT";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_APPEAL_SUCCESS"] = 12] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_APPEAL_SUCCESS";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_MATCH"] = 13] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_MATCH";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_LIVE_GOAL"] = 14] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_LIVE_GOAL";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_SUBSCRIPTION"] = 15] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_SUBSCRIPTION";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_STAR_COMMENT"] = 16] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_STAR_COMMENT";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_RANKING"] = 17] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_RANKING";
    PerceptionDialogIconType[PerceptionDialogIconType["PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_COMMON"] = 18] = "PERCEPTION_DIALOG_ICON_TYPE_ICON_TYPE_COMMON";
    PerceptionDialogIconType[PerceptionDialogIconType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PerceptionDialogIconType = exports.PerceptionDialogIconType || (exports.PerceptionDialogIconType = {}));
var GoalMessageSource;
(function (GoalMessageSource) {
    GoalMessageSource[GoalMessageSource["GOAL_MESSAGE_SOURCE_UNKNOWN"] = 0] = "GOAL_MESSAGE_SOURCE_UNKNOWN";
    GoalMessageSource[GoalMessageSource["GOAL_MESSAGE_SOURCE_COMMIT"] = 1] = "GOAL_MESSAGE_SOURCE_COMMIT";
    GoalMessageSource[GoalMessageSource["GOAL_MESSAGE_SOURCE_PROGRESS_UPDATE"] = 2] = "GOAL_MESSAGE_SOURCE_PROGRESS_UPDATE";
    GoalMessageSource[GoalMessageSource["GOAL_MESSAGE_SOURCE_PIN"] = 3] = "GOAL_MESSAGE_SOURCE_PIN";
    GoalMessageSource[GoalMessageSource["GOAL_MESSAGE_SOURCE_UNPIN"] = 4] = "GOAL_MESSAGE_SOURCE_UNPIN";
    GoalMessageSource[GoalMessageSource["GOAL_MESSAGE_SOURCE_REVIEW_CALLBACK"] = 5] = "GOAL_MESSAGE_SOURCE_REVIEW_CALLBACK";
    GoalMessageSource[GoalMessageSource["GOAL_MESSAGE_SOURCE_SUSPEND"] = 6] = "GOAL_MESSAGE_SOURCE_SUSPEND";
    GoalMessageSource[GoalMessageSource["GOAL_MESSAGE_SOURCE_CHALLENGE_PROMPT"] = 7] = "GOAL_MESSAGE_SOURCE_CHALLENGE_PROMPT";
    GoalMessageSource[GoalMessageSource["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GoalMessageSource = exports.GoalMessageSource || (exports.GoalMessageSource = {}));
var ExhibitionType;
(function (ExhibitionType) {
    ExhibitionType[ExhibitionType["EXHIBITION_TYPE_DEFAULT"] = 0] = "EXHIBITION_TYPE_DEFAULT";
    ExhibitionType[ExhibitionType["EXHIBITION_TYPE_FOLD"] = 1] = "EXHIBITION_TYPE_FOLD";
    ExhibitionType[ExhibitionType["EXHIBITION_TYPE_PUBLIC_SCREEN"] = 2] = "EXHIBITION_TYPE_PUBLIC_SCREEN";
    ExhibitionType[ExhibitionType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ExhibitionType = exports.ExhibitionType || (exports.ExhibitionType = {}));
var GiftSource;
(function (GiftSource) {
    GiftSource[GiftSource["GIFT_SOURCE_UNKNOWN"] = 0] = "GIFT_SOURCE_UNKNOWN";
    GiftSource[GiftSource["GIFT_SOURCE_PLATFORM"] = 1] = "GIFT_SOURCE_PLATFORM";
    GiftSource[GiftSource["GIFT_SOURCE_USER_BUY_RANDOM"] = 2] = "GIFT_SOURCE_USER_BUY_RANDOM";
    GiftSource[GiftSource["GIFT_SOURCE_USER_BUY_SPECIFIC"] = 3] = "GIFT_SOURCE_USER_BUY_SPECIFIC";
    GiftSource[GiftSource["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GiftSource = exports.GiftSource || (exports.GiftSource = {}));
var MessageDisplayStyle;
(function (MessageDisplayStyle) {
    MessageDisplayStyle[MessageDisplayStyle["MESSAGE_DISPLAY_STYLE_DEFAULT"] = 0] = "MESSAGE_DISPLAY_STYLE_DEFAULT";
    MessageDisplayStyle[MessageDisplayStyle["MESSAGE_DISPLAY_STYLE_POPUP"] = 1] = "MESSAGE_DISPLAY_STYLE_POPUP";
    MessageDisplayStyle[MessageDisplayStyle["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(MessageDisplayStyle = exports.MessageDisplayStyle || (exports.MessageDisplayStyle = {}));
var ProfitRankType;
(function (ProfitRankType) {
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_HOURLY_RANK"] = 0] = "PROFIT_RANK_TYPE_TYPE_HOURLY_RANK";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_WEEKLY_RANK"] = 1] = "PROFIT_RANK_TYPE_TYPE_WEEKLY_RANK";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_HOURLY_STAR_RANK"] = 2] = "PROFIT_RANK_TYPE_TYPE_HOURLY_STAR_RANK";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_WEEKLY_RISING_RANK_ACTIVITY"] = 3] = "PROFIT_RANK_TYPE_TYPE_WEEKLY_RISING_RANK_ACTIVITY";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_WEEKLY_RISING_RANK"] = 4] = "PROFIT_RANK_TYPE_TYPE_WEEKLY_RISING_RANK";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_WEEKLY_ROOKIE"] = 5] = "PROFIT_RANK_TYPE_TYPE_WEEKLY_ROOKIE";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_E_COMMERCE_WEEKLY"] = 6] = "PROFIT_RANK_TYPE_TYPE_E_COMMERCE_WEEKLY";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_E_COMMERCE_DAILY"] = 7] = "PROFIT_RANK_TYPE_TYPE_E_COMMERCE_DAILY";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_DAILY_RANK"] = 8] = "PROFIT_RANK_TYPE_TYPE_DAILY_RANK";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_FIRST_GIFT_RANK"] = 9] = "PROFIT_RANK_TYPE_TYPE_FIRST_GIFT_RANK";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_GAME_RANK"] = 10] = "PROFIT_RANK_TYPE_TYPE_GAME_RANK";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_DAILY_GAME"] = 11] = "PROFIT_RANK_TYPE_TYPE_DAILY_GAME";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_HALL_OF_FAME_RANK"] = 12] = "PROFIT_RANK_TYPE_TYPE_HALL_OF_FAME_RANK";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_RANK_LEAGUE"] = 13] = "PROFIT_RANK_TYPE_TYPE_RANK_LEAGUE";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_DAILY_ROOKIE"] = 14] = "PROFIT_RANK_TYPE_DAILY_ROOKIE";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_TEAM_RANK"] = 15] = "PROFIT_RANK_TYPE_TYPE_TEAM_RANK";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_CLASS_RANK"] = 16] = "PROFIT_RANK_TYPE_TYPE_CLASS_RANK";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_DAILY_GAME_PUB_G"] = 20] = "PROFIT_RANK_TYPE_TYPE_DAILY_GAME_PUB_G";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_DAILY_GAME_MLB_B"] = 21] = "PROFIT_RANK_TYPE_TYPE_DAILY_GAME_MLB_B";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_DAILY_GAME_FREE_FIRE"] = 22] = "PROFIT_RANK_TYPE_TYPE_DAILY_GAME_FREE_FIRE";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_WEEKLY_GAME_SUBCATEGORY_ONE"] = 23] = "PROFIT_RANK_TYPE_TYPE_WEEKLY_GAME_SUBCATEGORY_ONE";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_WEEKLY_GAME_SUBCATEGORY_TWO"] = 24] = "PROFIT_RANK_TYPE_TYPE_WEEKLY_GAME_SUBCATEGORY_TWO";
    ProfitRankType[ProfitRankType["PROFIT_RANK_TYPE_TYPE_WEEKLY_GAME_SUBCATEGORY_THREE"] = 25] = "PROFIT_RANK_TYPE_TYPE_WEEKLY_GAME_SUBCATEGORY_THREE";
    ProfitRankType[ProfitRankType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ProfitRankType = exports.ProfitRankType || (exports.ProfitRankType = {}));
var UnionAnimationInfoType;
(function (UnionAnimationInfoType) {
    UnionAnimationInfoType[UnionAnimationInfoType["UNION_ANIMATION_INFO_TYPE_NO_UNION_ANIMATION"] = 0] = "UNION_ANIMATION_INFO_TYPE_NO_UNION_ANIMATION";
    UnionAnimationInfoType[UnionAnimationInfoType["UNION_ANIMATION_INFO_TYPE_LOOP"] = 1] = "UNION_ANIMATION_INFO_TYPE_LOOP";
    UnionAnimationInfoType[UnionAnimationInfoType["UNION_ANIMATION_INFO_TYPE_LOCK"] = 2] = "UNION_ANIMATION_INFO_TYPE_LOCK";
    UnionAnimationInfoType[UnionAnimationInfoType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(UnionAnimationInfoType = exports.UnionAnimationInfoType || (exports.UnionAnimationInfoType = {}));
var DisplayStyle;
(function (DisplayStyle) {
    DisplayStyle[DisplayStyle["DISPLAY_STYLE_NORMAL"] = 0] = "DISPLAY_STYLE_NORMAL";
    DisplayStyle[DisplayStyle["DISPLAY_STYLE_STAY"] = 1] = "DISPLAY_STYLE_STAY";
    DisplayStyle[DisplayStyle["DISPLAY_STYLE_CHAT"] = 2] = "DISPLAY_STYLE_CHAT";
    DisplayStyle[DisplayStyle["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(DisplayStyle = exports.DisplayStyle || (exports.DisplayStyle = {}));
var HitABStatus;
(function (HitABStatus) {
    HitABStatus[HitABStatus["HIT_A_B_STATUS_HIT_AB_STATUS_NO_HIT"] = 0] = "HIT_A_B_STATUS_HIT_AB_STATUS_NO_HIT";
    HitABStatus[HitABStatus["HIT_A_B_STATUS_HIT_AB_STATUS_ENTER_FROM_EXTERNAL_LINK_NEW_TEXT"] = 1] = "HIT_A_B_STATUS_HIT_AB_STATUS_ENTER_FROM_EXTERNAL_LINK_NEW_TEXT";
    HitABStatus[HitABStatus["HIT_A_B_STATUS_HIT_AB_STATUS_ENTER_FROM_RE_POST_NEW_TEXT"] = 2] = "HIT_A_B_STATUS_HIT_AB_STATUS_ENTER_FROM_RE_POST_NEW_TEXT";
    HitABStatus[HitABStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(HitABStatus = exports.HitABStatus || (exports.HitABStatus = {}));
var PollKind;
(function (PollKind) {
    PollKind[PollKind["POLL_KIND_NORMAL"] = 0] = "POLL_KIND_NORMAL";
    PollKind[PollKind["POLL_KIND_GIFT"] = 1] = "POLL_KIND_GIFT";
    PollKind[PollKind["POLL_KIND_CUSTOMIZABLE"] = 2] = "POLL_KIND_CUSTOMIZABLE";
    PollKind[PollKind["POLL_KIND_CUSTOMIZABLE_GIFT"] = 3] = "POLL_KIND_CUSTOMIZABLE_GIFT";
    PollKind[PollKind["POLL_KIND_QUICK_GIFT"] = 4] = "POLL_KIND_QUICK_GIFT";
    PollKind[PollKind["POLL_KIND_EMOTE"] = 5] = "POLL_KIND_EMOTE";
    PollKind[PollKind["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PollKind = exports.PollKind || (exports.PollKind = {}));
var PollTemplateStatus;
(function (PollTemplateStatus) {
    PollTemplateStatus[PollTemplateStatus["POLL_TEMPLATE_STATUS_TO_BE_REVIEWED"] = 0] = "POLL_TEMPLATE_STATUS_TO_BE_REVIEWED";
    PollTemplateStatus[PollTemplateStatus["POLL_TEMPLATE_STATUS_UNDER_REVIEW"] = 1] = "POLL_TEMPLATE_STATUS_UNDER_REVIEW";
    PollTemplateStatus[PollTemplateStatus["POLL_TEMPLATE_STATUS_REVIEWED"] = 2] = "POLL_TEMPLATE_STATUS_REVIEWED";
    PollTemplateStatus[PollTemplateStatus["POLL_TEMPLATE_STATUS_REFUSED"] = 3] = "POLL_TEMPLATE_STATUS_REFUSED";
    PollTemplateStatus[PollTemplateStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PollTemplateStatus = exports.PollTemplateStatus || (exports.PollTemplateStatus = {}));
var PollAppealStatus;
(function (PollAppealStatus) {
    PollAppealStatus[PollAppealStatus["POLL_APPEAL_STATUS_UNKNOWN"] = 0] = "POLL_APPEAL_STATUS_UNKNOWN";
    PollAppealStatus[PollAppealStatus["POLL_APPEAL_STATUS_PASS"] = 1] = "POLL_APPEAL_STATUS_PASS";
    PollAppealStatus[PollAppealStatus["POLL_APPEAL_STATUS_FAIL"] = 2] = "POLL_APPEAL_STATUS_FAIL";
    PollAppealStatus[PollAppealStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PollAppealStatus = exports.PollAppealStatus || (exports.PollAppealStatus = {}));
var RankTestMessageScene;
(function (RankTestMessageScene) {
    RankTestMessageScene[RankTestMessageScene["RANK_TEST_MESSAGE_SCENE_UNKNOWN"] = 0] = "RANK_TEST_MESSAGE_SCENE_UNKNOWN";
    RankTestMessageScene[RankTestMessageScene["RANK_TEST_MESSAGE_SCENE_ONLINE_AUDIENCE_TOP_N_UPDATE_PUBLIC_SCREEN"] = 1] = "RANK_TEST_MESSAGE_SCENE_ONLINE_AUDIENCE_TOP_N_UPDATE_PUBLIC_SCREEN";
    RankTestMessageScene[RankTestMessageScene["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(RankTestMessageScene = exports.RankTestMessageScene || (exports.RankTestMessageScene = {}));
var TriggerReason;
(function (TriggerReason) {
    TriggerReason[TriggerReason["TRIGGER_REASON_UNKNOWN"] = 0] = "TRIGGER_REASON_UNKNOWN";
    TriggerReason[TriggerReason["TRIGGER_REASON_SCORE_UPDATE"] = 1] = "TRIGGER_REASON_SCORE_UPDATE";
    TriggerReason[TriggerReason["TRIGGER_REASON_BATTLE_END"] = 2] = "TRIGGER_REASON_BATTLE_END";
    TriggerReason[TriggerReason["TRIGGER_REASON_OPT_OUT_UPDATE"] = 3] = "TRIGGER_REASON_OPT_OUT_UPDATE";
    TriggerReason[TriggerReason["TRIGGER_REASON_KEEP_ALIVE"] = 4] = "TRIGGER_REASON_KEEP_ALIVE";
    TriggerReason[TriggerReason["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(TriggerReason = exports.TriggerReason || (exports.TriggerReason = {}));
var Reason;
(function (Reason) {
    Reason[Reason["REASON_TIME_UP"] = 0] = "REASON_TIME_UP";
    Reason[Reason["REASON_CUT_SHORT"] = 1] = "REASON_CUT_SHORT";
    Reason[Reason["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Reason = exports.Reason || (exports.Reason = {}));
var BattleTaskMessageType;
(function (BattleTaskMessageType) {
    BattleTaskMessageType[BattleTaskMessageType["BATTLE_TASK_MESSAGE_TYPE_START"] = 0] = "BATTLE_TASK_MESSAGE_TYPE_START";
    BattleTaskMessageType[BattleTaskMessageType["BATTLE_TASK_MESSAGE_TYPE_TASK_UPDATE"] = 1] = "BATTLE_TASK_MESSAGE_TYPE_TASK_UPDATE";
    BattleTaskMessageType[BattleTaskMessageType["BATTLE_TASK_MESSAGE_TYPE_TASK_SETTLE"] = 2] = "BATTLE_TASK_MESSAGE_TYPE_TASK_SETTLE";
    BattleTaskMessageType[BattleTaskMessageType["BATTLE_TASK_MESSAGE_TYPE_REWARD_SETTLE"] = 3] = "BATTLE_TASK_MESSAGE_TYPE_REWARD_SETTLE";
    BattleTaskMessageType[BattleTaskMessageType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BattleTaskMessageType = exports.BattleTaskMessageType || (exports.BattleTaskMessageType = {}));
var RewardStatus;
(function (RewardStatus) {
    RewardStatus[RewardStatus["REWARD_STATUS_SUCCEED"] = 0] = "REWARD_STATUS_SUCCEED";
    RewardStatus[RewardStatus["REWARD_STATUS_FAILED"] = 1] = "REWARD_STATUS_FAILED";
    RewardStatus[RewardStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(RewardStatus = exports.RewardStatus || (exports.RewardStatus = {}));
var BattleAction;
(function (BattleAction) {
    BattleAction[BattleAction["BATTLE_ACTION_UNKNOWN"] = 0] = "BATTLE_ACTION_UNKNOWN";
    BattleAction[BattleAction["BATTLE_ACTION_INVITE"] = 1] = "BATTLE_ACTION_INVITE";
    BattleAction[BattleAction["BATTLE_ACTION_REJECT"] = 2] = "BATTLE_ACTION_REJECT";
    BattleAction[BattleAction["BATTLE_ACTION_CANCEL"] = 3] = "BATTLE_ACTION_CANCEL";
    BattleAction[BattleAction["BATTLE_ACTION_OPEN"] = 4] = "BATTLE_ACTION_OPEN";
    BattleAction[BattleAction["BATTLE_ACTION_FINISH"] = 5] = "BATTLE_ACTION_FINISH";
    BattleAction[BattleAction["BATTLE_ACTION_CUT_SHORT"] = 6] = "BATTLE_ACTION_CUT_SHORT";
    BattleAction[BattleAction["BATTLE_ACTION_ACCEPT"] = 7] = "BATTLE_ACTION_ACCEPT";
    BattleAction[BattleAction["BATTLE_ACTION_QUIT_APPLY"] = 8] = "BATTLE_ACTION_QUIT_APPLY";
    BattleAction[BattleAction["BATTLE_ACTION_DECLINE_QUIT"] = 9] = "BATTLE_ACTION_DECLINE_QUIT";
    BattleAction[BattleAction["BATTLE_ACTION_DECLINE_OFF_QUIT"] = 10] = "BATTLE_ACTION_DECLINE_OFF_QUIT";
    BattleAction[BattleAction["BATTLE_ACTION_LEAVE_LINK_MIC"] = 11] = "BATTLE_ACTION_LEAVE_LINK_MIC";
    BattleAction[BattleAction["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BattleAction = exports.BattleAction || (exports.BattleAction = {}));
var Result;
(function (Result) {
    Result[Result["RESULT_WIN"] = 0] = "RESULT_WIN";
    Result[Result["RESULT_LOSE"] = 1] = "RESULT_LOSE";
    Result[Result["RESULT_DRAW"] = 2] = "RESULT_DRAW";
    Result[Result["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Result = exports.Result || (exports.Result = {}));
var GiftPermissionType;
(function (GiftPermissionType) {
    GiftPermissionType[GiftPermissionType["GIFT_PERMISSION_TYPE_UNKNOWN_TYPE"] = 0] = "GIFT_PERMISSION_TYPE_UNKNOWN_TYPE";
    GiftPermissionType[GiftPermissionType["GIFT_PERMISSION_TYPE_NO_GIFT_PERMISSION"] = 1] = "GIFT_PERMISSION_TYPE_NO_GIFT_PERMISSION";
    GiftPermissionType[GiftPermissionType["GIFT_PERMISSION_TYPE_ANCHOR_CLOSE"] = 2] = "GIFT_PERMISSION_TYPE_ANCHOR_CLOSE";
    GiftPermissionType[GiftPermissionType["GIFT_PERMISSION_TYPE_HAS_GIFT_PERMISSION"] = 3] = "GIFT_PERMISSION_TYPE_HAS_GIFT_PERMISSION";
    GiftPermissionType[GiftPermissionType["GIFT_PERMISSION_TYPE_ANCHOR_BANNED"] = 4] = "GIFT_PERMISSION_TYPE_ANCHOR_BANNED";
    GiftPermissionType[GiftPermissionType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GiftPermissionType = exports.GiftPermissionType || (exports.GiftPermissionType = {}));
var BattleABTestType;
(function (BattleABTestType) {
    BattleABTestType[BattleABTestType["BATTLE_A_B_TEST_TYPE_UNKNOWN_AB_TEST_TYPE"] = 0] = "BATTLE_A_B_TEST_TYPE_UNKNOWN_AB_TEST_TYPE";
    BattleABTestType[BattleABTestType["BATTLE_A_B_TEST_TYPE_MEANWHILE_INVITE"] = 1] = "BATTLE_A_B_TEST_TYPE_MEANWHILE_INVITE";
    BattleABTestType[BattleABTestType["BATTLE_A_B_TEST_TYPE_SPECIFIED_GIFT"] = 2] = "BATTLE_A_B_TEST_TYPE_SPECIFIED_GIFT";
    BattleABTestType[BattleABTestType["BATTLE_A_B_TEST_TYPE_RT_C_MESSAGE_CHANNEL"] = 3] = "BATTLE_A_B_TEST_TYPE_RT_C_MESSAGE_CHANNEL";
    BattleABTestType[BattleABTestType["BATTLE_A_B_TEST_TYPE_CONNECTION_TIME_OUT"] = 4] = "BATTLE_A_B_TEST_TYPE_CONNECTION_TIME_OUT";
    BattleABTestType[BattleABTestType["BATTLE_A_B_TEST_TYPE_REMATCH_SKIP_TEAMMATE"] = 5] = "BATTLE_A_B_TEST_TYPE_REMATCH_SKIP_TEAMMATE";
    BattleABTestType[BattleABTestType["BATTLE_A_B_TEST_TYPE_OPT_INVITEE_4048"] = 6] = "BATTLE_A_B_TEST_TYPE_OPT_INVITEE_4048";
    BattleABTestType[BattleABTestType["BATTLE_A_B_TEST_TYPE_BATTLE_AB_TEST_TYPE_TIME_CALIBRATE"] = 7] = "BATTLE_A_B_TEST_TYPE_BATTLE_AB_TEST_TYPE_TIME_CALIBRATE";
    BattleABTestType[BattleABTestType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BattleABTestType = exports.BattleABTestType || (exports.BattleABTestType = {}));
var PlayScene;
(function (PlayScene) {
    PlayScene[PlayScene["PLAY_SCENE_UNKNOWN"] = 0] = "PLAY_SCENE_UNKNOWN";
    PlayScene[PlayScene["PLAY_SCENE_COUNTDOWN_FOR_ALL"] = 1] = "PLAY_SCENE_COUNTDOWN_FOR_ALL";
    PlayScene[PlayScene["PLAY_SCENE_COUNTDOWN_FOR_SINGLE"] = 2] = "PLAY_SCENE_COUNTDOWN_FOR_SINGLE";
    PlayScene[PlayScene["PLAY_SCENE_LIVE_SHOW"] = 3] = "PLAY_SCENE_LIVE_SHOW";
    PlayScene[PlayScene["PLAY_SCENE_AIG_C"] = 4] = "PLAY_SCENE_AIG_C";
    PlayScene[PlayScene["PLAY_SCENE_KARAOKE"] = 5] = "PLAY_SCENE_KARAOKE";
    PlayScene[PlayScene["PLAY_SCENE_DRAW_GUESS"] = 6] = "PLAY_SCENE_DRAW_GUESS";
    PlayScene[PlayScene["PLAY_SCENE_ENLARGE_GRID"] = 7] = "PLAY_SCENE_ENLARGE_GRID";
    PlayScene[PlayScene["PLAY_SCENE_GIFT_PRIORITY_LINK"] = 8] = "PLAY_SCENE_GIFT_PRIORITY_LINK";
    PlayScene[PlayScene["PLAY_SCENE_GIFT_THRESHOLD_LINK"] = 9] = "PLAY_SCENE_GIFT_THRESHOLD_LINK";
    PlayScene[PlayScene["PLAY_SCENE_NOTICE_BOARD"] = 10] = "PLAY_SCENE_NOTICE_BOARD";
    PlayScene[PlayScene["PLAY_SCENE_PLAY_BOOK"] = 11] = "PLAY_SCENE_PLAY_BOOK";
    PlayScene[PlayScene["PLAY_SCENE_GUEST_SHOWDOWN"] = 12] = "PLAY_SCENE_GUEST_SHOWDOWN";
    PlayScene[PlayScene["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PlayScene = exports.PlayScene || (exports.PlayScene = {}));
var LinkType;
(function (LinkType) {
    LinkType[LinkType["LINK_TYPE_TYPE_UNKNOWN"] = 0] = "LINK_TYPE_TYPE_UNKNOWN";
    LinkType[LinkType["LINK_TYPE_TYPE_VIDEO"] = 1] = "LINK_TYPE_TYPE_VIDEO";
    LinkType[LinkType["LINK_TYPE_TYPE_AUDIO"] = 2] = "LINK_TYPE_TYPE_AUDIO";
    LinkType[LinkType["LINK_TYPE_TYPE_VIRTUAL"] = 3] = "LINK_TYPE_TYPE_VIRTUAL";
    LinkType[LinkType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkType = exports.LinkType || (exports.LinkType = {}));
var LinkSilenceStatus;
(function (LinkSilenceStatus) {
    LinkSilenceStatus[LinkSilenceStatus["LINK_SILENCE_STATUS_STATUS_UN_SILENCE"] = 0] = "LINK_SILENCE_STATUS_STATUS_UN_SILENCE";
    LinkSilenceStatus[LinkSilenceStatus["LINK_SILENCE_STATUS_STATUS_SILENCE_BY_SELF"] = 1] = "LINK_SILENCE_STATUS_STATUS_SILENCE_BY_SELF";
    LinkSilenceStatus[LinkSilenceStatus["LINK_SILENCE_STATUS_STATUS_SILENCE_BY_OWNER"] = 2] = "LINK_SILENCE_STATUS_STATUS_SILENCE_BY_OWNER";
    LinkSilenceStatus[LinkSilenceStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkSilenceStatus = exports.LinkSilenceStatus || (exports.LinkSilenceStatus = {}));
var LinkmicRoleType;
(function (LinkmicRoleType) {
    LinkmicRoleType[LinkmicRoleType["LINKMIC_ROLE_TYPE_ROLE_TYPE_UNKOWN"] = 0] = "LINKMIC_ROLE_TYPE_ROLE_TYPE_UNKOWN";
    LinkmicRoleType[LinkmicRoleType["LINKMIC_ROLE_TYPE_LEADER"] = 1] = "LINKMIC_ROLE_TYPE_LEADER";
    LinkmicRoleType[LinkmicRoleType["LINKMIC_ROLE_TYPE_PLAYER"] = 2] = "LINKMIC_ROLE_TYPE_PLAYER";
    LinkmicRoleType[LinkmicRoleType["LINKMIC_ROLE_TYPE_INVITEE"] = 3] = "LINKMIC_ROLE_TYPE_INVITEE";
    LinkmicRoleType[LinkmicRoleType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkmicRoleType = exports.LinkmicRoleType || (exports.LinkmicRoleType = {}));
var LinkRoleType;
(function (LinkRoleType) {
    LinkRoleType[LinkRoleType["LINK_ROLE_TYPE_TYPE_ROLE_TYPE_UNKOWN"] = 0] = "LINK_ROLE_TYPE_TYPE_ROLE_TYPE_UNKOWN";
    LinkRoleType[LinkRoleType["LINK_ROLE_TYPE_TYPE_LEADER"] = 1] = "LINK_ROLE_TYPE_TYPE_LEADER";
    LinkRoleType[LinkRoleType["LINK_ROLE_TYPE_TYPE_PLAYER"] = 2] = "LINK_ROLE_TYPE_TYPE_PLAYER";
    LinkRoleType[LinkRoleType["LINK_ROLE_TYPE_TYPE_INVITEE"] = 3] = "LINK_ROLE_TYPE_TYPE_INVITEE";
    LinkRoleType[LinkRoleType["LINK_ROLE_TYPE_TYPE_APPLIER"] = 4] = "LINK_ROLE_TYPE_TYPE_APPLIER";
    LinkRoleType[LinkRoleType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkRoleType = exports.LinkRoleType || (exports.LinkRoleType = {}));
var MuteStatus;
(function (MuteStatus) {
    MuteStatus[MuteStatus["MUTE_STATUS_MUTE"] = 0] = "MUTE_STATUS_MUTE";
    MuteStatus[MuteStatus["MUTE_STATUS_UN_MUTE"] = 1] = "MUTE_STATUS_UN_MUTE";
    MuteStatus[MuteStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(MuteStatus = exports.MuteStatus || (exports.MuteStatus = {}));
var GuestMicCameraManageOp;
(function (GuestMicCameraManageOp) {
    GuestMicCameraManageOp[GuestMicCameraManageOp["GUEST_MIC_CAMERA_MANAGE_OP_OPEN_MIC"] = 0] = "GUEST_MIC_CAMERA_MANAGE_OP_OPEN_MIC";
    GuestMicCameraManageOp[GuestMicCameraManageOp["GUEST_MIC_CAMERA_MANAGE_OP_OPEN_CAMERA"] = 1] = "GUEST_MIC_CAMERA_MANAGE_OP_OPEN_CAMERA";
    GuestMicCameraManageOp[GuestMicCameraManageOp["GUEST_MIC_CAMERA_MANAGE_OP_CLOSE_MIC"] = 2] = "GUEST_MIC_CAMERA_MANAGE_OP_CLOSE_MIC";
    GuestMicCameraManageOp[GuestMicCameraManageOp["GUEST_MIC_CAMERA_MANAGE_OP_CLOSE_CAMERA"] = 3] = "GUEST_MIC_CAMERA_MANAGE_OP_CLOSE_CAMERA";
    GuestMicCameraManageOp[GuestMicCameraManageOp["GUEST_MIC_CAMERA_MANAGE_OP_CLOSE_MIC_PUNISH"] = 4] = "GUEST_MIC_CAMERA_MANAGE_OP_CLOSE_MIC_PUNISH";
    GuestMicCameraManageOp[GuestMicCameraManageOp["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GuestMicCameraManageOp = exports.GuestMicCameraManageOp || (exports.GuestMicCameraManageOp = {}));
var GuestMicCameraChangeScene;
(function (GuestMicCameraChangeScene) {
    GuestMicCameraChangeScene[GuestMicCameraChangeScene["GUEST_MIC_CAMERA_CHANGE_SCENE_CHANGE_SCENE_UNKNOWN"] = 0] = "GUEST_MIC_CAMERA_CHANGE_SCENE_CHANGE_SCENE_UNKNOWN";
    GuestMicCameraChangeScene[GuestMicCameraChangeScene["GUEST_MIC_CAMERA_CHANGE_SCENE_LIVE_SHOW_BY_ANCHOR_AUTO"] = 1] = "GUEST_MIC_CAMERA_CHANGE_SCENE_LIVE_SHOW_BY_ANCHOR_AUTO";
    GuestMicCameraChangeScene[GuestMicCameraChangeScene["GUEST_MIC_CAMERA_CHANGE_SCENE_LIVE_SHOW_BY_SERVER_NORMAL"] = 2] = "GUEST_MIC_CAMERA_CHANGE_SCENE_LIVE_SHOW_BY_SERVER_NORMAL";
    GuestMicCameraChangeScene[GuestMicCameraChangeScene["GUEST_MIC_CAMERA_CHANGE_SCENE_LIVE_SHOW_BY_SHOW_END"] = 3] = "GUEST_MIC_CAMERA_CHANGE_SCENE_LIVE_SHOW_BY_SHOW_END";
    GuestMicCameraChangeScene[GuestMicCameraChangeScene["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GuestMicCameraChangeScene = exports.GuestMicCameraChangeScene || (exports.GuestMicCameraChangeScene = {}));
var LinkMicUserAdminType;
(function (LinkMicUserAdminType) {
    LinkMicUserAdminType[LinkMicUserAdminType["LINK_MIC_USER_ADMIN_TYPE_UNDEFINED_TYPE"] = 0] = "LINK_MIC_USER_ADMIN_TYPE_UNDEFINED_TYPE";
    LinkMicUserAdminType[LinkMicUserAdminType["LINK_MIC_USER_ADMIN_TYPE_MANAGER_TYPE"] = 1] = "LINK_MIC_USER_ADMIN_TYPE_MANAGER_TYPE";
    LinkMicUserAdminType[LinkMicUserAdminType["LINK_MIC_USER_ADMIN_TYPE_HOST_TYPE"] = 2] = "LINK_MIC_USER_ADMIN_TYPE_HOST_TYPE";
    LinkMicUserAdminType[LinkMicUserAdminType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkMicUserAdminType = exports.LinkMicUserAdminType || (exports.LinkMicUserAdminType = {}));
var LinkmicMultiLiveEnum;
(function (LinkmicMultiLiveEnum) {
    LinkmicMultiLiveEnum[LinkmicMultiLiveEnum["LINKMIC_MULTI_LIVE_ENUM_DEFAULT"] = 0] = "LINKMIC_MULTI_LIVE_ENUM_DEFAULT";
    LinkmicMultiLiveEnum[LinkmicMultiLiveEnum["LINKMIC_MULTI_LIVE_ENUM_ANCHOR_USE_NEW_LAYOUT"] = 1] = "LINKMIC_MULTI_LIVE_ENUM_ANCHOR_USE_NEW_LAYOUT";
    LinkmicMultiLiveEnum[LinkmicMultiLiveEnum["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkmicMultiLiveEnum = exports.LinkmicMultiLiveEnum || (exports.LinkmicMultiLiveEnum = {}));
var PollEndType;
(function (PollEndType) {
    PollEndType[PollEndType["POLL_END_TYPE_POLL_END_BY_TIME"] = 0] = "POLL_END_TYPE_POLL_END_BY_TIME";
    PollEndType[PollEndType["POLL_END_TYPE_POLL_END_BY_OWNER"] = 1] = "POLL_END_TYPE_POLL_END_BY_OWNER";
    PollEndType[PollEndType["POLL_END_TYPE_POLL_END_BY_OTHER"] = 2] = "POLL_END_TYPE_POLL_END_BY_OTHER";
    PollEndType[PollEndType["POLL_END_TYPE_POLL_END_BY_ADMIN"] = 3] = "POLL_END_TYPE_POLL_END_BY_ADMIN";
    PollEndType[PollEndType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PollEndType = exports.PollEndType || (exports.PollEndType = {}));
var CohostABTestType;
(function (CohostABTestType) {
    CohostABTestType[CohostABTestType["COHOST_A_B_TEST_TYPE_COHOST_AB_TEST_TYPE_UNKNOWN"] = 0] = "COHOST_A_B_TEST_TYPE_COHOST_AB_TEST_TYPE_UNKNOWN";
    CohostABTestType[CohostABTestType["COHOST_A_B_TEST_TYPE_COHOST_AB_TEST_TYPE_LINK_TIME_OUT_STRATEGY"] = 1] = "COHOST_A_B_TEST_TYPE_COHOST_AB_TEST_TYPE_LINK_TIME_OUT_STRATEGY";
    CohostABTestType[CohostABTestType["COHOST_A_B_TEST_TYPE_COHOST_AB_TEST_TYPE_COHOST_RESERVATION"] = 2] = "COHOST_A_B_TEST_TYPE_COHOST_AB_TEST_TYPE_COHOST_RESERVATION";
    CohostABTestType[CohostABTestType["COHOST_A_B_TEST_TYPE_COHOST_AB_TEST_TYPE_QUICK_PAIR_NEW_ARCH_SWITCH"] = 3] = "COHOST_A_B_TEST_TYPE_COHOST_AB_TEST_TYPE_QUICK_PAIR_NEW_ARCH_SWITCH";
    CohostABTestType[CohostABTestType["COHOST_A_B_TEST_TYPE_COHOST_AB_TEST_TYPE_COHOST_INVITATION_TEXT"] = 4] = "COHOST_A_B_TEST_TYPE_COHOST_AB_TEST_TYPE_COHOST_INVITATION_TEXT";
    CohostABTestType[CohostABTestType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(CohostABTestType = exports.CohostABTestType || (exports.CohostABTestType = {}));
var OptPairStatus;
(function (OptPairStatus) {
    OptPairStatus[OptPairStatus["OPT_PAIR_STATUS_UNKNOWN"] = 0] = "OPT_PAIR_STATUS_UNKNOWN";
    OptPairStatus[OptPairStatus["OPT_PAIR_STATUS_OFFLINE"] = 1] = "OPT_PAIR_STATUS_OFFLINE";
    OptPairStatus[OptPairStatus["OPT_PAIR_STATUS_FINISHED"] = 2] = "OPT_PAIR_STATUS_FINISHED";
    OptPairStatus[OptPairStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(OptPairStatus = exports.OptPairStatus || (exports.OptPairStatus = {}));
var ContentPositionType;
(function (ContentPositionType) {
    ContentPositionType[ContentPositionType["CONTENT_POSITION_TYPE_UNKNOWN"] = 0] = "CONTENT_POSITION_TYPE_UNKNOWN";
    ContentPositionType[ContentPositionType["CONTENT_POSITION_TYPE_STREAM"] = 1] = "CONTENT_POSITION_TYPE_STREAM";
    ContentPositionType[ContentPositionType["CONTENT_POSITION_TYPE_LIVE_STUDIO_STREAM_PORTRAIT"] = 2] = "CONTENT_POSITION_TYPE_LIVE_STUDIO_STREAM_PORTRAIT";
    ContentPositionType[ContentPositionType["CONTENT_POSITION_TYPE_LIVE_STUDIO_STREAM_LANDSCAPE"] = 3] = "CONTENT_POSITION_TYPE_LIVE_STUDIO_STREAM_LANDSCAPE";
    ContentPositionType[ContentPositionType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ContentPositionType = exports.ContentPositionType || (exports.ContentPositionType = {}));
var MultiGuestOutsideRoomInviteSource;
(function (MultiGuestOutsideRoomInviteSource) {
    MultiGuestOutsideRoomInviteSource[MultiGuestOutsideRoomInviteSource["MULTI_GUEST_OUTSIDE_ROOM_INVITE_SOURCE_OUTSIDE_ROOM_INVITE_SOURCE_UNKNOWN"] = 0] = "MULTI_GUEST_OUTSIDE_ROOM_INVITE_SOURCE_OUTSIDE_ROOM_INVITE_SOURCE_UNKNOWN";
    MultiGuestOutsideRoomInviteSource[MultiGuestOutsideRoomInviteSource["MULTI_GUEST_OUTSIDE_ROOM_INVITE_SOURCE_OUTSIDE_ROOM_INVITE_SOURCE_PANEL"] = 1] = "MULTI_GUEST_OUTSIDE_ROOM_INVITE_SOURCE_OUTSIDE_ROOM_INVITE_SOURCE_PANEL";
    MultiGuestOutsideRoomInviteSource[MultiGuestOutsideRoomInviteSource["MULTI_GUEST_OUTSIDE_ROOM_INVITE_SOURCE_OUTSIDE_ROOM_INVITE_SOURCE_CAPSULE"] = 2] = "MULTI_GUEST_OUTSIDE_ROOM_INVITE_SOURCE_OUTSIDE_ROOM_INVITE_SOURCE_CAPSULE";
    MultiGuestOutsideRoomInviteSource[MultiGuestOutsideRoomInviteSource["MULTI_GUEST_OUTSIDE_ROOM_INVITE_SOURCE_OUTSIDE_ROOM_INVITE_SOURCE_EMPTY_POSITION"] = 3] = "MULTI_GUEST_OUTSIDE_ROOM_INVITE_SOURCE_OUTSIDE_ROOM_INVITE_SOURCE_EMPTY_POSITION";
    MultiGuestOutsideRoomInviteSource[MultiGuestOutsideRoomInviteSource["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(MultiGuestOutsideRoomInviteSource = exports.MultiGuestOutsideRoomInviteSource || (exports.MultiGuestOutsideRoomInviteSource = {}));
var LinkUserType;
(function (LinkUserType) {
    LinkUserType[LinkUserType["LINK_USER_TYPE_DEFAULT"] = 0] = "LINK_USER_TYPE_DEFAULT";
    LinkUserType[LinkUserType["LINK_USER_TYPE_KARAOKE"] = 1] = "LINK_USER_TYPE_KARAOKE";
    LinkUserType[LinkUserType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkUserType = exports.LinkUserType || (exports.LinkUserType = {}));
var ContentInviteSource;
(function (ContentInviteSource) {
    ContentInviteSource[ContentInviteSource["CONTENT_INVITE_SOURCE_INVITE_SOURCE_UNKNOWN"] = 0] = "CONTENT_INVITE_SOURCE_INVITE_SOURCE_UNKNOWN";
    ContentInviteSource[ContentInviteSource["CONTENT_INVITE_SOURCE_INVITE_SOURCE_PANEL_GO_LIVE"] = 1] = "CONTENT_INVITE_SOURCE_INVITE_SOURCE_PANEL_GO_LIVE";
    ContentInviteSource[ContentInviteSource["CONTENT_INVITE_SOURCE_INVITE_SOURCE_MUTUAL_NOTICE"] = 2] = "CONTENT_INVITE_SOURCE_INVITE_SOURCE_MUTUAL_NOTICE";
    ContentInviteSource[ContentInviteSource["CONTENT_INVITE_SOURCE_INVITE_SOURCE_USER_PROFILE"] = 3] = "CONTENT_INVITE_SOURCE_INVITE_SOURCE_USER_PROFILE";
    ContentInviteSource[ContentInviteSource["CONTENT_INVITE_SOURCE_INVITE_SOURCE_RESERVE"] = 4] = "CONTENT_INVITE_SOURCE_INVITE_SOURCE_RESERVE";
    ContentInviteSource[ContentInviteSource["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ContentInviteSource = exports.ContentInviteSource || (exports.ContentInviteSource = {}));
var LinkmicShareRevenueSetting;
(function (LinkmicShareRevenueSetting) {
    LinkmicShareRevenueSetting[LinkmicShareRevenueSetting["LINKMIC_SHARE_REVENUE_SETTING_LINK_MIC_SHARE_REVENUE_NOT_SET"] = 0] = "LINKMIC_SHARE_REVENUE_SETTING_LINK_MIC_SHARE_REVENUE_NOT_SET";
    LinkmicShareRevenueSetting[LinkmicShareRevenueSetting["LINKMIC_SHARE_REVENUE_SETTING_LINK_MIC_SHARE_REVENUE_OPEN"] = 1] = "LINKMIC_SHARE_REVENUE_SETTING_LINK_MIC_SHARE_REVENUE_OPEN";
    LinkmicShareRevenueSetting[LinkmicShareRevenueSetting["LINKMIC_SHARE_REVENUE_SETTING_LINK_MIC_SHARE_REVENUE_CLOSE"] = 2] = "LINKMIC_SHARE_REVENUE_SETTING_LINK_MIC_SHARE_REVENUE_CLOSE";
    LinkmicShareRevenueSetting[LinkmicShareRevenueSetting["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(LinkmicShareRevenueSetting = exports.LinkmicShareRevenueSetting || (exports.LinkmicShareRevenueSetting = {}));
var PosIdentityType;
(function (PosIdentityType) {
    PosIdentityType[PosIdentityType["POS_IDENTITY_TYPE_IDENTITY_EMPTY_SLOT"] = 0] = "POS_IDENTITY_TYPE_IDENTITY_EMPTY_SLOT";
    PosIdentityType[PosIdentityType["POS_IDENTITY_TYPE_IDENTITY_RT_C_USER_ID"] = 1] = "POS_IDENTITY_TYPE_IDENTITY_RT_C_USER_ID";
    PosIdentityType[PosIdentityType["POS_IDENTITY_TYPE_IDENTITY_RT_C_STREAM_ID"] = 2] = "POS_IDENTITY_TYPE_IDENTITY_RT_C_STREAM_ID";
    PosIdentityType[PosIdentityType["POS_IDENTITY_TYPE_IDENTITY_LIVE_USER_ID"] = 3] = "POS_IDENTITY_TYPE_IDENTITY_LIVE_USER_ID";
    PosIdentityType[PosIdentityType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PosIdentityType = exports.PosIdentityType || (exports.PosIdentityType = {}));
var JoinType;
(function (JoinType) {
    JoinType[JoinType["JOIN_TYPE_UNKNOWN"] = 0] = "JOIN_TYPE_UNKNOWN";
    JoinType[JoinType["JOIN_TYPE_CHANNEL_APPLY"] = 1] = "JOIN_TYPE_CHANNEL_APPLY";
    JoinType[JoinType["JOIN_TYPE_CHANNEL_INVITE"] = 2] = "JOIN_TYPE_CHANNEL_INVITE";
    JoinType[JoinType["JOIN_TYPE_GROUP_APPLY"] = 100] = "JOIN_TYPE_GROUP_APPLY";
    JoinType[JoinType["JOIN_TYPE_GROUP_APPLY_FOLLOW"] = 101] = "JOIN_TYPE_GROUP_APPLY_FOLLOW";
    JoinType[JoinType["JOIN_TYPE_GROUP_INVITE"] = 102] = "JOIN_TYPE_GROUP_INVITE";
    JoinType[JoinType["JOIN_TYPE_GROUP_INVITE_FOLLOW"] = 103] = "JOIN_TYPE_GROUP_INVITE_FOLLOW";
    JoinType[JoinType["JOIN_TYPE_GROUP_OWNER_JOIN"] = 104] = "JOIN_TYPE_GROUP_OWNER_JOIN";
    JoinType[JoinType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(JoinType = exports.JoinType || (exports.JoinType = {}));
var CohostLayoutMode;
(function (CohostLayoutMode) {
    CohostLayoutMode[CohostLayoutMode["COHOST_LAYOUT_MODE_NORMAL"] = 0] = "COHOST_LAYOUT_MODE_NORMAL";
    CohostLayoutMode[CohostLayoutMode["COHOST_LAYOUT_MODE_SCREEN_SHARE"] = 1] = "COHOST_LAYOUT_MODE_SCREEN_SHARE";
    CohostLayoutMode[CohostLayoutMode["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(CohostLayoutMode = exports.CohostLayoutMode || (exports.CohostLayoutMode = {}));
var TagClassification;
(function (TagClassification) {
    TagClassification[TagClassification["TAG_CLASSIFICATION_UNKNOWN"] = 0] = "TAG_CLASSIFICATION_UNKNOWN";
    TagClassification[TagClassification["TAG_CLASSIFICATION_COHOST_HISTORY"] = 1] = "TAG_CLASSIFICATION_COHOST_HISTORY";
    TagClassification[TagClassification["TAG_CLASSIFICATION_FIRST_DEGREE_RELATION"] = 2] = "TAG_CLASSIFICATION_FIRST_DEGREE_RELATION";
    TagClassification[TagClassification["TAG_CLASSIFICATION_SECOND_DEGREE_RELATION"] = 3] = "TAG_CLASSIFICATION_SECOND_DEGREE_RELATION";
    TagClassification[TagClassification["TAG_CLASSIFICATION_RANK"] = 4] = "TAG_CLASSIFICATION_RANK";
    TagClassification[TagClassification["TAG_CLASSIFICATION_SIMILAR_INTERESTS"] = 5] = "TAG_CLASSIFICATION_SIMILAR_INTERESTS";
    TagClassification[TagClassification["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(TagClassification = exports.TagClassification || (exports.TagClassification = {}));
var SourceType;
(function (SourceType) {
    SourceType[SourceType["SOURCE_TYPE_UNKNOWN"] = 0] = "SOURCE_TYPE_UNKNOWN";
    SourceType[SourceType["SOURCE_TYPE_FRIEND_LIST"] = 1] = "SOURCE_TYPE_FRIEND_LIST";
    SourceType[SourceType["SOURCE_TYPE_RECOMMEND_LIST"] = 2] = "SOURCE_TYPE_RECOMMEND_LIST";
    SourceType[SourceType["SOURCE_TYPE_RECENT"] = 3] = "SOURCE_TYPE_RECENT";
    SourceType[SourceType["SOURCE_TYPE_OTHER_FOLLOW"] = 4] = "SOURCE_TYPE_OTHER_FOLLOW";
    SourceType[SourceType["SOURCE_TYPE_QUICK_PAIR"] = 5] = "SOURCE_TYPE_QUICK_PAIR";
    SourceType[SourceType["SOURCE_TYPE_ACTIVITY"] = 6] = "SOURCE_TYPE_ACTIVITY";
    SourceType[SourceType["SOURCE_TYPE_QUICK_RECOMMEND"] = 7] = "SOURCE_TYPE_QUICK_RECOMMEND";
    SourceType[SourceType["SOURCE_TYPE_OFFICIAL_CHANNEL"] = 8] = "SOURCE_TYPE_OFFICIAL_CHANNEL";
    SourceType[SourceType["SOURCE_TYPE_BEST_TEAMMATE"] = 9] = "SOURCE_TYPE_BEST_TEAMMATE";
    SourceType[SourceType["SOURCE_TYPE_RESERVATION"] = 10] = "SOURCE_TYPE_RESERVATION";
    SourceType[SourceType["SOURCE_TYPE_PAIRING"] = 11] = "SOURCE_TYPE_PAIRING";
    SourceType[SourceType["SOURCE_TYPE_PAIRING_ON_RESERVATION"] = 12] = "SOURCE_TYPE_PAIRING_ON_RESERVATION";
    SourceType[SourceType["SOURCE_TYPE_TOPIC_QUICK_PAIR"] = 13] = "SOURCE_TYPE_TOPIC_QUICK_PAIR";
    SourceType[SourceType["SOURCE_TYPE_TOPIC_QUICK_RECOMMEND"] = 14] = "SOURCE_TYPE_TOPIC_QUICK_RECOMMEND";
    SourceType[SourceType["SOURCE_TYPE_ONLINE_FRIEND_CAPSULE"] = 15] = "SOURCE_TYPE_ONLINE_FRIEND_CAPSULE";
    SourceType[SourceType["SOURCE_TYPE_WEEKLY_RANK"] = 20] = "SOURCE_TYPE_WEEKLY_RANK";
    SourceType[SourceType["SOURCE_TYPE_HOURLY_RANK"] = 21] = "SOURCE_TYPE_HOURLY_RANK";
    SourceType[SourceType["SOURCE_TYPE_WEEKLY_RISING"] = 23] = "SOURCE_TYPE_WEEKLY_RISING";
    SourceType[SourceType["SOURCE_TYPE_WEEKLY_ROOKIE"] = 24] = "SOURCE_TYPE_WEEKLY_ROOKIE";
    SourceType[SourceType["SOURCE_TYPE_CONNECTION_LIST"] = 25] = "SOURCE_TYPE_CONNECTION_LIST";
    SourceType[SourceType["SOURCE_TYPE_DAILY_RANK"] = 26] = "SOURCE_TYPE_DAILY_RANK";
    SourceType[SourceType["SOURCE_TYPE_DAILY_RANK_HALL_OF_FAME"] = 27] = "SOURCE_TYPE_DAILY_RANK_HALL_OF_FAME";
    SourceType[SourceType["SOURCE_TYPE_RESERVATION_BUBBLE"] = 28] = "SOURCE_TYPE_RESERVATION_BUBBLE";
    SourceType[SourceType["SOURCE_TYPE_PAIRING_BUBBLE"] = 29] = "SOURCE_TYPE_PAIRING_BUBBLE";
    SourceType[SourceType["SOURCE_TYPE_LEAGUE_PHASE_ONE"] = 30] = "SOURCE_TYPE_LEAGUE_PHASE_ONE";
    SourceType[SourceType["SOURCE_TYPE_LEAGUE_PHASE_TWO"] = 31] = "SOURCE_TYPE_LEAGUE_PHASE_TWO";
    SourceType[SourceType["SOURCE_TYPE_LEAGUE_PHASE_THREE"] = 32] = "SOURCE_TYPE_LEAGUE_PHASE_THREE";
    SourceType[SourceType["SOURCE_TYPE_DAILY_ROOKIE"] = 33] = "SOURCE_TYPE_DAILY_ROOKIE";
    SourceType[SourceType["SOURCE_TYPE_MAY_KNOW_LIST"] = 34] = "SOURCE_TYPE_MAY_KNOW_LIST";
    SourceType[SourceType["SOURCE_TYPE_BANNER"] = 35] = "SOURCE_TYPE_BANNER";
    SourceType[SourceType["SOURCE_TYPE_FANS_TEAM_RANK"] = 36] = "SOURCE_TYPE_FANS_TEAM_RANK";
    SourceType[SourceType["SOURCE_TYPE_SEARCH"] = 37] = "SOURCE_TYPE_SEARCH";
    SourceType[SourceType["SOURCE_TYPE_E_OY_RANK_LIST"] = 38] = "SOURCE_TYPE_E_OY_RANK_LIST";
    SourceType[SourceType["SOURCE_TYPE_LEAGUE_CAMPAIGN_RANK"] = 39] = "SOURCE_TYPE_LEAGUE_CAMPAIGN_RANK";
    SourceType[SourceType["SOURCE_TYPE_CREATOR_CLASS_RANK"] = 40] = "SOURCE_TYPE_CREATOR_CLASS_RANK";
    SourceType[SourceType["SOURCE_TYPE_HISTORY"] = 41] = "SOURCE_TYPE_HISTORY";
    SourceType[SourceType["SOURCE_TYPE_QUICK_RECOMMEND_DURING_COHOST"] = 43] = "SOURCE_TYPE_QUICK_RECOMMEND_DURING_COHOST";
    SourceType[SourceType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(SourceType = exports.SourceType || (exports.SourceType = {}));
var BattleType;
(function (BattleType) {
    BattleType[BattleType["BATTLE_TYPE_UNKNOWN_BATTLE_TYPE"] = 0] = "BATTLE_TYPE_UNKNOWN_BATTLE_TYPE";
    BattleType[BattleType["BATTLE_TYPE_NORMAL_BATTLE"] = 1] = "BATTLE_TYPE_NORMAL_BATTLE";
    BattleType[BattleType["BATTLE_TYPE_TEAM_BATTLE"] = 2] = "BATTLE_TYPE_TEAM_BATTLE";
    BattleType[BattleType["BATTLE_TYPE_INDIVIDUAL_BATTLE"] = 3] = "BATTLE_TYPE_INDIVIDUAL_BATTLE";
    BattleType[BattleType["BATTLE_TYPE_1_V_N"] = 4] = "BATTLE_TYPE_1_V_N";
    BattleType[BattleType["BATTLE_TYPE_TAKE_THE_STAGE"] = 51] = "BATTLE_TYPE_TAKE_THE_STAGE";
    BattleType[BattleType["BATTLE_TYPE_GROUP_SHOW"] = 52] = "BATTLE_TYPE_GROUP_SHOW";
    BattleType[BattleType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BattleType = exports.BattleType || (exports.BattleType = {}));
var BattleInviteType;
(function (BattleInviteType) {
    BattleInviteType[BattleInviteType["BATTLE_INVITE_TYPE_NORMAL"] = 0] = "BATTLE_INVITE_TYPE_NORMAL";
    BattleInviteType[BattleInviteType["BATTLE_INVITE_TYPE_AGAIN"] = 1] = "BATTLE_INVITE_TYPE_AGAIN";
    BattleInviteType[BattleInviteType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BattleInviteType = exports.BattleInviteType || (exports.BattleInviteType = {}));
